diff --git a/gfx/layers/moz.build b/gfx/layers/moz.build
index dcee9ad9e38a..f681304fe7ea 100644
--- a/gfx/layers/moz.build
+++ b/gfx/layers/moz.build
@@ -280,6 +280,7 @@ if CONFIG["MOZ_WIDGET_TOOLKIT"] in ("cocoa", "uikit"):
     ]
     UNIFIED_SOURCES += [
         "NativeLayerCA.mm",
+        "NativeLayerCAPictureInPicture.mm",
         "NativeLayerMacSurfaceHandler.mm",
         "NativeLayerRemoteMac.mm",
         "NativeLayerRootRemoteMacChild.mm",
