diff --git a/gfx/layers/moz.build b/gfx/layers/moz.build
index 5c22be217f56..5cb1bced6c3a 100644
--- a/gfx/layers/moz.build
+++ b/gfx/layers/moz.build
@@ -280,6 +280,7 @@ if CONFIG["MOZ_WIDGET_TOOLKIT"] in ("cocoa", "uikit"):
     ]
     UNIFIED_SOURCES += [
         "NativeLayerCA.mm",
+        "NativeLayerCAPictureInPicture.mm",
         "NativeLayerMacSurfaceHandler.mm",
         "NativeLayerRemoteMac.mm",
         "NativeLayerRootRemoteMacChild.mm",
