diff --git a/gfx/wgpu_bindings/src/server.rs b/gfx/wgpu_bindings/src/server.rs
index ecd988c4a428..9d33d7d7b8b7 100644
--- a/gfx/wgpu_bindings/src/server.rs
+++ b/gfx/wgpu_bindings/src/server.rs
@@ -59,7 +59,12 @@ const MAX_TEXTURE_EXTENT: u32 = std::i16::MAX as u32;
 // Then we may have to ignore the spec or get it changed.
 const MAX_BINDINGS_PER_RESOURCE_TYPE: u32 = 64;
 
-#[cfg(any(target_os = "windows", target_os = "linux", target_os = "macos"))]
+#[cfg(any(
+    target_os = "windows",
+    target_os = "linux",
+    target_os = "macos",
+    target_os = "ios"
+))]
 fn emit_critical_invalid_note(what: &'static str) {
     // SAFETY: We ensure that the pointer provided is not null.
     let msg = CString::new(format!("{what} is invalid")).unwrap();
@@ -3448,31 +3453,39 @@ pub unsafe extern "C" fn wgpu_server_device_wait_fence_from_shared_handle(
     res.is_ok()
 }
 
-#[cfg(target_os = "macos")]
+#[cfg(any(target_os = "macos", target_os = "ios"))]
 mod macos {
+    #[cfg(target_os = "macos")]
     use std::ffi::CString;
 
-    use super::{emit_critical_invalid_note, gfx_critical_note, Global};
+    use super::{emit_critical_invalid_note, Global};
+    #[cfg(target_os = "macos")]
+    use super::gfx_critical_note;
+    use crate::{error::ErrorBuffer, wgpu_string};
+    #[cfg(target_os = "macos")]
     use crate::{
-        error::ErrorBuffer,
         server::{
             wgpu_server_ensure_shared_texture_for_swap_chain,
             wgpu_server_get_external_io_surface_id,
         },
-        wgpu_string, SwapChainId,
+        SwapChainId,
     };
 
     use nsstring::nsACString;
+    #[cfg(target_os = "macos")]
     use objc2::{
         rc::{autoreleasepool, Retained},
         runtime::ProtocolObject,
     };
+    #[cfg(target_os = "macos")]
     use objc2_foundation::NSString;
     use objc2_io_surface::IOSurfaceRef;
     use objc2_metal::{
-        MTLDevice as _, MTLPixelFormat, MTLResource, MTLStorageMode, MTLTexture,
-        MTLTextureDescriptor, MTLTextureType, MTLTextureUsage,
+        MTLDevice as _, MTLPixelFormat, MTLStorageMode, MTLTextureDescriptor,
+        MTLTextureType, MTLTextureUsage,
     };
+    #[cfg(target_os = "macos")]
+    use objc2_metal::{MTLResource, MTLTexture};
     use wgc::id;
 
     /// Imports a Metal texture from the specified plane of an IOSurface.
@@ -3562,6 +3575,7 @@ mod macos {
         }
     }
 
+    #[cfg(target_os = "macos")]
     impl super::Global {
         pub(super) fn create_texture_with_shared_texture_iosurface(
             &self,
