diff --git a/ipc/chromium/src/base/message_pump_kqueue.cc b/ipc/chromium/src/base/message_pump_kqueue.cc
index 7159a223f4b0..83fd22a0422d 100644
--- a/ipc/chromium/src/base/message_pump_kqueue.cc
+++ b/ipc/chromium/src/base/message_pump_kqueue.cc
@@ -6,10 +6,6 @@
 
 #include <sys/errno.h>
 
-#ifdef XP_IOS
-#  include <BrowserEngineCore/BEkevent.h>
-#endif
-
 #include "mozilla/AutoRestore.h"
 
 #include "base/logging.h"
@@ -20,15 +16,9 @@ namespace base {
 
 namespace {
 
-// On iOS, the normal `kevent64` method is blocked by the content process
-// sandbox, so instead we use `be_kevent64` from the BrowserEngineCore library.
 static int platform_kevent64(int fd, const kevent64_s* changelist, int nchanges,
                              kevent64_s* eventlist, int nevents, int flags) {
-#ifdef XP_IOS
-  return be_kevent64(fd, changelist, nchanges, eventlist, nevents, flags);
-#else
   return kevent64(fd, changelist, nchanges, eventlist, nevents, flags, nullptr);
-#endif
 }
 
 int ChangeOneEvent(const mozilla::UniqueFileHandle& kqueue, kevent64_s* event) {
