diff --git a/dom/webidl/Navigator.webidl b/dom/webidl/Navigator.webidl
index 00c1d6089082..e2334e2d472a 100644
--- a/dom/webidl/Navigator.webidl
+++ b/dom/webidl/Navigator.webidl
@@ -48,7 +48,7 @@ interface mixin NavigatorID {
   readonly attribute DOMString appCodeName; // constant "Mozilla"
   [Constant, Cached]
   readonly attribute DOMString appName; // constant "Netscape"
-  [Constant, Cached, Throws, NeedsCallerType]
+  [Pure, Cached, Throws, NeedsCallerType]
   readonly attribute DOMString appVersion;
   [Pure, Cached, Throws, NeedsCallerType]
   readonly attribute DOMString platform;
@@ -174,7 +174,7 @@ partial interface Navigator {
 };
 
 partial interface Navigator {
-  [Throws, Constant, Cached, NeedsCallerType]
+  [Throws, Pure, Cached, NeedsCallerType]
   readonly attribute DOMString oscpu;
   // WebKit/Blink support this; Trident/Presto do not.
   readonly attribute DOMString vendor;
@@ -184,7 +184,7 @@ partial interface Navigator {
   readonly attribute DOMString productSub;
   // WebKit/Blink/Trident/Presto support this.
   readonly attribute boolean cookieEnabled;
-  [Throws, Constant, Cached, NeedsCallerType]
+  [Throws, Pure, Cached, NeedsCallerType]
   readonly attribute DOMString buildID;
 
   // WebKit/Blink/Trident/Presto support this.
