diff --git a/toolkit/components/extensions/ExtensionChild.sys.mjs b/toolkit/components/extensions/ExtensionChild.sys.mjs
index f0c8b871036e..facabed0ad3c 100644
--- a/toolkit/components/extensions/ExtensionChild.sys.mjs
+++ b/toolkit/components/extensions/ExtensionChild.sys.mjs
@@ -215,7 +215,11 @@ export class MessageEvent extends SimpleEventAPI {
 }
 
 function holdMessage(name, anonymizedName, data, native = null) {
-  if (native && AppConstants.platform !== "android") {
+  if (
+    native &&
+    AppConstants.platform !== "android" &&
+    AppConstants.platform !== "ios"
+  ) {
     data = lazy.NativeApp.encodeMessage(native.context, data);
   }
   return new StructuredCloneHolder(name, anonymizedName, data);
