diff --git a/mobile/android/components/geckoview/components.conf b/mobile/android/components/geckoview/components.conf
index 165cad582a8a..889a409990f3 100644
--- a/mobile/android/components/geckoview/components.conf
+++ b/mobile/android/components/geckoview/components.conf
@@ -8,9 +8,9 @@ Classes = [
         'contract_ids': [
             '@mozilla.org/uriloader/external-helper-app-service;1',
         ],
-        'type': 'GeckoViewExternalAppService',
-        'constructor': 'GeckoViewExternalAppService::GetSingleton',
-        'headers': ['GeckoViewExternalAppService.h'],
+        'type': 'GeckoViewExternalResponseService',
+        'constructor': 'GeckoViewExternalResponseService::GetSingleton',
+        'headers': ['GeckoViewExternalResponseService.h'],
         'processes': ProcessSelector.ALLOW_IN_SOCKET_PROCESS,
     },
     {
