diff --git a/mobile/android/components/geckoview/GeckoViewExternalAppService.cpp b/mobile/android/components/geckoview/GeckoViewExternalAppService.cpp
index cd614ddb25aa..43e78c992bab 100644
--- a/mobile/android/components/geckoview/GeckoViewExternalAppService.cpp
+++ b/mobile/android/components/geckoview/GeckoViewExternalAppService.cpp
@@ -2,7 +2,7 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-#include "GeckoViewExternalAppService.h"
+#include "GeckoViewExternalResponseService.h"
 
 #include "mozilla/dom/BrowsingContext.h"
 #include "mozilla/dom/CanonicalBrowsingContext.h"
@@ -36,24 +36,24 @@ class StreamListener final : public mozilla::GeckoViewStreamListener {
   RefPtr<nsWindow> mWindow;
 };
 
-mozilla::StaticRefPtr<GeckoViewExternalAppService>
-    GeckoViewExternalAppService::sService;
+mozilla::StaticRefPtr<GeckoViewExternalResponseService>
+    GeckoViewExternalResponseService::sService;
 
 /* static */
-already_AddRefed<GeckoViewExternalAppService>
-GeckoViewExternalAppService::GetSingleton() {
+already_AddRefed<GeckoViewExternalResponseService>
+GeckoViewExternalResponseService::GetSingleton() {
   if (!sService) {
-    sService = new GeckoViewExternalAppService();
+    sService = new GeckoViewExternalResponseService();
   }
-  RefPtr<GeckoViewExternalAppService> service = sService;
+  RefPtr<GeckoViewExternalResponseService> service = sService;
   return service.forget();
 }
 
-GeckoViewExternalAppService::GeckoViewExternalAppService() {}
+GeckoViewExternalResponseService::GeckoViewExternalResponseService() {}
 
-NS_IMPL_ISUPPORTS(GeckoViewExternalAppService, nsIExternalHelperAppService);
+NS_IMPL_ISUPPORTS(GeckoViewExternalResponseService, nsIExternalHelperAppService);
 
-NS_IMETHODIMP GeckoViewExternalAppService::DoContent(
+NS_IMETHODIMP GeckoViewExternalResponseService::DoContent(
     const nsACString& aMimeContentType, nsIChannel* aChannel,
     nsIInterfaceRequestor* aContentContext, bool aForceSave,
     nsIInterfaceRequestor* aWindowContext,
@@ -61,7 +61,7 @@ NS_IMETHODIMP GeckoViewExternalAppService::DoContent(
   return NS_ERROR_FAILURE;
 }
 
-NS_IMETHODIMP GeckoViewExternalAppService::CreateListener(
+NS_IMETHODIMP GeckoViewExternalResponseService::CreateListener(
     const nsACString& aMimeContentType, nsIChannel* aChannel,
     mozilla::dom::BrowsingContext* aContentContext, bool aForceSave,
     nsIInterfaceRequestor* aWindowContext,
@@ -92,7 +92,7 @@ NS_IMETHODIMP GeckoViewExternalAppService::CreateListener(
   return NS_OK;
 }
 
-NS_IMETHODIMP GeckoViewExternalAppService::ApplyDecodingForExtension(
+NS_IMETHODIMP GeckoViewExternalResponseService::ApplyDecodingForExtension(
     const nsACString& aExtension, const nsACString& aEncodingType,
     bool* aApplyDecoding) {
   // This currently doesn't matter, because we never read the stream.
@@ -100,7 +100,7 @@ NS_IMETHODIMP GeckoViewExternalAppService::ApplyDecodingForExtension(
   return NS_OK;
 }
 
-NS_IMETHODIMP GeckoViewExternalAppService::GetPreferredDownloadsDirectory(
+NS_IMETHODIMP GeckoViewExternalResponseService::GetPreferredDownloadsDirectory(
     nsIFile** _retval) {
   MOZ_ASSERT(false);
   return nsresult::NS_ERROR_NOT_IMPLEMENTED;
