diff --git a/mobile/android/components/geckoview/GeckoViewExternalAppService.h b/mobile/android/components/geckoview/GeckoViewExternalAppService.h
index 63f07716b750..58296377a5e9 100644
--- a/mobile/android/components/geckoview/GeckoViewExternalAppService.h
+++ b/mobile/android/components/geckoview/GeckoViewExternalAppService.h
@@ -2,24 +2,24 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-#ifndef GeckoViewExternalAppService_h_
-#define GeckoViewExternalAppService_h_
+#ifndef GeckoViewExternalResponseService_h_
+#define GeckoViewExternalResponseService_h_
 
 #include "nsIExternalHelperAppService.h"
 #include "mozilla/StaticPtr.h"
 
-class GeckoViewExternalAppService : public nsIExternalHelperAppService {
+class GeckoViewExternalResponseService : public nsIExternalHelperAppService {
  public:
   NS_DECL_ISUPPORTS
   NS_DECL_NSIEXTERNALHELPERAPPSERVICE
 
-  GeckoViewExternalAppService();
+  GeckoViewExternalResponseService();
 
-  static already_AddRefed<GeckoViewExternalAppService> GetSingleton();
+  static already_AddRefed<GeckoViewExternalResponseService> GetSingleton();
 
  private:
-  virtual ~GeckoViewExternalAppService() {}
-  static mozilla::StaticRefPtr<GeckoViewExternalAppService> sService;
+  virtual ~GeckoViewExternalResponseService() {}
+  static mozilla::StaticRefPtr<GeckoViewExternalResponseService> sService;
 };
 
 #endif
