diff --git a/mobile/android/components/geckoview/moz.build b/mobile/android/components/geckoview/moz.build
index ea294ea3dd3b..52204f4e8b22 100644
--- a/mobile/android/components/geckoview/moz.build
+++ b/mobile/android/components/geckoview/moz.build
@@ -7,7 +7,7 @@ UNIFIED_SOURCES += [
     "GeckoViewContentChannelChild.cpp",
     "GeckoViewContentChannelParent.cpp",
     "GeckoViewContentProtocolHandler.cpp",
-    "GeckoViewExternalAppService.cpp",
+    "GeckoViewExternalResponseService.cpp",
     "GeckoViewInputStream.cpp",
     "GeckoViewOutputStream.cpp",
     "GeckoViewStreamListener.cpp",
@@ -16,7 +16,7 @@ UNIFIED_SOURCES += [
 EXPORTS += [
     "GeckoViewContentChannel.h",
     "GeckoViewContentProtocolHandler.h",
-    "GeckoViewExternalAppService.h",
+    "GeckoViewExternalResponseService.h",
     "GeckoViewInputStream.h",
     "GeckoViewOutputStream.h",
     "GeckoViewStreamListener.h",
