diff --git a/widget/InputData.cpp b/widget/InputData.cpp
index 57b8f8cd4b90..4d90e562a680 100644
--- a/widget/InputData.cpp
+++ b/widget/InputData.cpp
@@ -507,6 +507,7 @@ PanGestureInput::PanGestureInput()
       mHandledByAPZ(false),
       mOverscrollBehaviorAllowsSwipe(false),
       mSimulateMomentum(false),
+      mAllowFlingAcceleration(true),
       mIsNoLineOrPageDelta(true),
       mMayTriggerSwipe(false) {}
 
@@ -525,6 +526,7 @@ PanGestureInput::PanGestureInput(PanGestureType aType, TimeStamp aTimeStamp,
       mHandledByAPZ(false),
       mOverscrollBehaviorAllowsSwipe(false),
       mSimulateMomentum(false),
+      mAllowFlingAcceleration(true),
       mIsNoLineOrPageDelta(true) {
   mMayTriggerSwipe = SwipeTracker::CanTriggerSwipe(*this);
 }
