diff --git a/widget/InputData.cpp b/widget/InputData.cpp
index 798cb5ecb139..14bceec7de71 100644
--- a/widget/InputData.cpp
+++ b/widget/InputData.cpp
@@ -496,6 +496,7 @@ PanGestureInput::PanGestureInput()
       mHandledByAPZ(false),
       mOverscrollBehaviorAllowsSwipe(false),
       mSimulateMomentum(false),
+      mAllowFlingAcceleration(true),
       mIsNoLineOrPageDelta(true),
       mMayTriggerSwipe(false) {}
 
@@ -514,6 +515,7 @@ PanGestureInput::PanGestureInput(PanGestureType aType, TimeStamp aTimeStamp,
       mHandledByAPZ(false),
       mOverscrollBehaviorAllowsSwipe(false),
       mSimulateMomentum(false),
+      mAllowFlingAcceleration(true),
       mIsNoLineOrPageDelta(true) {
   mMayTriggerSwipe = SwipeTracker::CanTriggerSwipe(*this);
 }
