diff --git a/widget/nsGUIEventIPC.h b/widget/nsGUIEventIPC.h
index ad2a71caa9b4..bca33a6dc50c 100644
--- a/widget/nsGUIEventIPC.h
+++ b/widget/nsGUIEventIPC.h
@@ -1044,6 +1044,7 @@ struct ParamTraits<mozilla::ContentCache> {
     WriteParam(aWriter, aParam.mCaret);
     WriteParam(aWriter, aParam.mTextRectArray);
     WriteParam(aWriter, aParam.mLastCommitStringTextRectArray);
+    WriteParam(aWriter, aParam.mEditorRectIsEditable);
     WriteParam(aWriter, aParam.mEditorRect);
   }
 
@@ -1055,6 +1056,7 @@ struct ParamTraits<mozilla::ContentCache> {
            ReadParam(aReader, &aResult->mCaret) &&
            ReadParam(aReader, &aResult->mTextRectArray) &&
            ReadParam(aReader, &aResult->mLastCommitStringTextRectArray) &&
+           ReadParam(aReader, &aResult->mEditorRectIsEditable) &&
            ReadParam(aReader, &aResult->mEditorRect);
   }
 };
