diff --git a/widget/nsGUIEventIPC.h b/widget/nsGUIEventIPC.h
index baf960fdb593..4fd872fb0889 100644
--- a/widget/nsGUIEventIPC.h
+++ b/widget/nsGUIEventIPC.h
@@ -1149,6 +1149,7 @@ struct ParamTraits<mozilla::ContentCache> {
     WriteParam(aWriter, aParam.mCaret);
     WriteParam(aWriter, aParam.mTextRectArray);
     WriteParam(aWriter, aParam.mLastCommitStringTextRectArray);
+    WriteParam(aWriter, aParam.mEditorRectIsEditable);
     WriteParam(aWriter, aParam.mEditorRect);
   }
 
@@ -1160,6 +1161,7 @@ struct ParamTraits<mozilla::ContentCache> {
            ReadParam(aReader, &aResult->mCaret) &&
            ReadParam(aReader, &aResult->mTextRectArray) &&
            ReadParam(aReader, &aResult->mLastCommitStringTextRectArray) &&
+           ReadParam(aReader, &aResult->mEditorRectIsEditable) &&
            ReadParam(aReader, &aResult->mEditorRect);
   }
 };
