diff --git a/widget/nsIWidget.h b/widget/nsIWidget.h
index 3690e1a7b5e9..ec2f1df90b50 100644
--- a/widget/nsIWidget.h
+++ b/widget/nsIWidget.h
@@ -2321,12 +2321,12 @@ class nsIWidget : public nsSupportsWeakReference {
    * @param aMessage message being sent to Android UI thread.
    */
   virtual void RecvToolbarAnimatorMessageFromCompositor(int32_t aMessage) {}
+#endif
 
   virtual void UpdateDynamicToolbarMaxHeight(mozilla::ScreenIntCoord aHeight) {}
   virtual mozilla::ScreenIntCoord GetDynamicToolbarMaxHeight() const {
     return 0;
   }
-#endif
 
   void EnsureLocalesChangedObserver();
   virtual void LocalesChanged() {}
