diff --git a/layout/generic/nsIFrame.cpp b/layout/generic/nsIFrame.cpp
index 3e3a98c3795b..f13090bc3efe 100644
--- a/layout/generic/nsIFrame.cpp
+++ b/layout/generic/nsIFrame.cpp
@@ -4957,6 +4957,12 @@ bool nsIFrame::IsSelectable(StyleUserSelect* aSelectStyle) const {
 }
 
 bool nsIFrame::ShouldPaintNormalSelection() const {
+// REYNARD: Hide the selection highlight because we use
+// iOS's native selection highlight instead.
+#if defined(XP_IOS)
+  return false;
+#endif
+
   if (IsSelectable()) {
     // NOTE: Ideally, we should return false if display selection is "OFF".
     // However, here is a hot path at painting.  Therefore, it should be checked
