diff --git a/gfx/layers/apz/src/AsyncPanZoomController.h b/gfx/layers/apz/src/AsyncPanZoomController.h
index 6801a0eee4af..69721e941dc5 100644
--- a/gfx/layers/apz/src/AsyncPanZoomController.h
+++ b/gfx/layers/apz/src/AsyncPanZoomController.h
@@ -1944,6 +1944,7 @@ class AsyncPanZoomController {
  private:
   // The timestamp of the latest touch start event.
   TimeStamp mTouchStartTime;
+  Maybe<TimeStamp> mPanGestureMayStartTime;
   // The start time of the latest autoscroll animation.
   TimeStamp mAutoscrollStartTime;
   // Used for interpolating touch events that cross the touch-start
