diff --git a/gfx/layers/apz/src/AsyncPanZoomController.h b/gfx/layers/apz/src/AsyncPanZoomController.h
index 7e922201775f..fadd4c7ef2d0 100644
--- a/gfx/layers/apz/src/AsyncPanZoomController.h
+++ b/gfx/layers/apz/src/AsyncPanZoomController.h
@@ -1963,6 +1963,7 @@ class AsyncPanZoomController {
  private:
   // The timestamp of the latest touch start event.
   TimeStamp mTouchStartTime;
+  Maybe<TimeStamp> mPanGestureMayStartTime;
   // The start time of the latest autoscroll animation.
   TimeStamp mAutoscrollStartTime;
   // Used for interpolating touch events that cross the touch-start
