diff --git a/mobile/shared/components/geckoview/GeckoViewPermission.sys.mjs b/mobile/shared/components/geckoview/GeckoViewPermission.sys.mjs
index 31789c65d0aa..444cb623375e 100644
--- a/mobile/shared/components/geckoview/GeckoViewPermission.sys.mjs
+++ b/mobile/shared/components/geckoview/GeckoViewPermission.sys.mjs
@@ -145,7 +145,7 @@ export class GeckoViewPermission {
 
       if (allowOrDeny === Services.perms.ALLOW_ACTION) {
         // Ask for app permission after asking for content permission.
-        if (perm.type === "geolocation") {
+        if (perm.type === "geolocation" && Services.appinfo.OS === "Android") {
           const granted = await actor.getAppPermissions([
             PERM_ACCESS_FINE_LOCATION,
           ]);
@@ -164,6 +164,24 @@ export class GeckoViewPermission {
 
     const allow = allowOrDeny === Services.perms.ALLOW_ACTION;
 
+    if (perm.type === "device-sensors") {
+      const browserId = actor.browsingContext.top.browserId;
+      if (
+        browserId &&
+        (allowOrDeny === Services.perms.ALLOW_ACTION ||
+          allowOrDeny === Services.perms.DENY_ACTION)
+      ) {
+        Services.perms.addFromPrincipalForBrowser(
+          principal,
+          perm.type,
+          allowOrDeny,
+          browserId
+        );
+      }
+
+      return { allow };
+    }
+
     // The storage access code adds itself to the perm manager; no need for us to do it.
     if (perm.type === "storage-access") {
       if (allow) {
