diff --git a/mobile/shared/components/geckoview/GeckoViewStartup.sys.mjs b/mobile/shared/components/geckoview/GeckoViewStartup.sys.mjs
index d58e0276979b..0169373cff32 100644
--- a/mobile/shared/components/geckoview/GeckoViewStartup.sys.mjs
+++ b/mobile/shared/components/geckoview/GeckoViewStartup.sys.mjs
@@ -178,6 +178,11 @@ export class GeckoViewStartup {
             "GeckoView:GetPermissionsByURI",
             "GeckoView:SetPermission",
             "GeckoView:SetPermissionByURI",
+            "GeckoView:RemovePermission",
+            "GeckoView:RemovePermissionByURI",
+            "GeckoView:GetCookieBannerModeForDomain",
+            "GeckoView:SetCookieBannerModeForDomain",
+            "GeckoView:RemoveCookieBannerModeForDomain",
           ],
         });
 
@@ -336,6 +341,23 @@ export class GeckoViewStartup {
           ged: ["GeckoView:Autofill:GetAddressStructure"],
         });
 
+        if (
+          Services.appinfo.processType == Services.appinfo.PROCESS_TYPE_DEFAULT
+        ) {
+          // We don't register this using the LazyGetter because it needs to be
+          // ready before the first call to the listener is received.
+          lazy.EventDispatcher.instance.registerListener(
+            lazy.GeckoViewPreferences,
+            [
+              "GeckoView:Preferences:GetPref",
+              "GeckoView:Preferences:SetPref",
+              "GeckoView:Preferences:ClearPref",
+              "GeckoView:Preferences:RegisterObserver",
+              "GeckoView:Preferences:UnregisterObserver",
+            ]
+          );
+        }
+
         break;
       }
 
