diff --git a/modules/libpref/init/all.js b/modules/libpref/init/all.js
index 29ff922cb70d..c22ee4ddab81 100644
--- a/modules/libpref/init/all.js
+++ b/modules/libpref/init/all.js
@@ -66,6 +66,10 @@ pref("security.crash_tracking.js_load_1.maxCrashes", 1);
 
 pref("general.useragent.compatMode.firefox", false);
 
+// Reduce GeckoView logging noise because the default "Debug" level
+// just makes debugging near to impossible.
+pref("geckoview.logging", "Warn");
+
 pref("general.config.obscure_value", 13); // for MCD .cfg files
 
 // Whether middle button click with a modifier key starts to autoscroll or
@@ -1898,7 +1902,7 @@ pref("dom.ipc.processCount.privilegedabout", 1);
 pref("dom.ipc.processCount.privilegedmozilla", 1);
 
 // Maximum number of isolated content processes per-origin.
-#ifdef ANDROID
+#if defined(ANDROID) || defined(XP_IOS)
 pref("dom.ipc.processCount.webIsolated", 1);
 #else
 pref("dom.ipc.processCount.webIsolated", 4);
