diff --git a/modules/libpref/init/all.js b/modules/libpref/init/all.js
index b0734147fb4d..46b23adbc155 100644
--- a/modules/libpref/init/all.js
+++ b/modules/libpref/init/all.js
@@ -66,6 +66,10 @@ pref("security.crash_tracking.js_load_1.maxCrashes", 1);
 
 pref("general.useragent.compatMode.firefox", false);
 
+// Reduce GeckoView logging noise because the default "Debug" level
+// just makes debugging near to impossible.
+pref("geckoview.logging", "Warn");
+
 pref("general.config.obscure_value", 13); // for MCD .cfg files
 
 // Whether middle button click with a modifier key starts to autoscroll or
