diff --git a/modules/libpref/init/all.js b/modules/libpref/init/all.js
index d740a8988b27..83f5da3539a0 100644
--- a/modules/libpref/init/all.js
+++ b/modules/libpref/init/all.js
@@ -66,6 +66,10 @@ pref("security.crash_tracking.js_load_1.maxCrashes", 1);
 
 pref("general.useragent.compatMode.firefox", false);
 
+// Reduce GeckoView logging noise because the default "Debug" level
+// just makes debugging near to impossible.
+pref("geckoview.logging", "Warn");
+
 pref("general.config.obscure_value", 13); // for MCD .cfg files
 
 // Whether middle button click with a modifier key starts to autoscroll or
@@ -1885,7 +1889,7 @@ pref("dom.ipc.processCount.privilegedabout", 1);
 pref("dom.ipc.processCount.privilegedmozilla", 1);
 
 // Maximum number of isolated content processes per-origin.
-#ifdef ANDROID
+#if defined(ANDROID) || defined(XP_IOS)
 pref("dom.ipc.processCount.webIsolated", 1);
 #else
 pref("dom.ipc.processCount.webIsolated", 4);
@@ -2666,7 +2670,7 @@ pref("font.size.monospace.x-math", 13);
   pref("helpers.private_mailcap_file", "~/.mailcap");
 #endif // ANDROID
 
-#if !defined(ANDROID) && !defined(XP_MACOSX) && defined(XP_UNIX)
+#if !defined(ANDROID) && !defined(XP_IOS) && !defined(XP_MACOSX) && defined(XP_UNIX)
   // Handled differently under Mac/Windows
   pref("network.protocol-handler.warn-external.file", false);
   pref("browser.drag_out_of_frame_style", 1);
@@ -2847,7 +2851,7 @@ pref("font.size.monospace.x-math", 13);
   // snooper.  So, let's use true for its default value.
   pref("intl.ime.hack.uim.using_key_snooper", true);
 
-#endif // !ANDROID && !XP_MACOSX && XP_UNIX
+#endif // !ANDROID && !XP_IOS && !XP_MACOSX && XP_UNIX
 
 #if defined(ANDROID)
 
