diff --git a/xpcom/threads/TaskController.cpp b/xpcom/threads/TaskController.cpp
index f7376c328f7f..be1d72942357 100644
--- a/xpcom/threads/TaskController.cpp
+++ b/xpcom/threads/TaskController.cpp
@@ -60,7 +60,7 @@ int32_t TaskController::GetPoolThreadCount() {
   }
 
   int32_t numCores = 0;
-#if defined(XP_MACOSX) && defined(__aarch64__)
+#if defined(XP_IOS) || (defined(XP_MACOSX) && defined(__aarch64__))
   if (const auto& cpuInfo = hal::GetHeterogeneousCpuInfo()) {
     // -1 because of the main thread.
     numCores = cpuInfo->mBigCpus.Count() + cpuInfo->mMediumCpus.Count() - 1;
