diff --git a/xpcom/threads/TaskController.cpp b/xpcom/threads/TaskController.cpp
index 28a1f2cb30e1..42419daa8d5e 100644
--- a/xpcom/threads/TaskController.cpp
+++ b/xpcom/threads/TaskController.cpp
@@ -58,7 +58,7 @@ int32_t TaskController::GetPoolThreadCount() {
   }
 
   int32_t numCores = 0;
-#if defined(XP_MACOSX) && defined(__aarch64__)
+#if defined(XP_IOS) || (defined(XP_MACOSX) && defined(__aarch64__))
   if (const auto& cpuInfo = hal::GetHeterogeneousCpuInfo()) {
     // -1 because of the main thread.
     numCores = cpuInfo->mBigCpus.Count() + cpuInfo->mMediumCpus.Count() - 1;
