diff --git a/gfx/thebes/gfxFontEntry.cpp b/gfx/thebes/gfxFontEntry.cpp
index ff5b3496ed31..053fc35ed9e9 100644
--- a/gfx/thebes/gfxFontEntry.cpp
+++ b/gfx/thebes/gfxFontEntry.cpp
@@ -1620,9 +1620,11 @@ void gfxFontFamily::LocalizedName(nsACString& aLocalizedName) {
 void gfxFontFamily::FindFontForChar(GlobalFontMatch* aMatchData) {
   gfxPlatformFontList::PlatformFontList()->mLock.AssertCurrentThreadIn();
 
+  const bool prefersColor = PrefersColor(aMatchData->mPresentation);
   {
     AutoReadLock lock(mLock);
-    if (mFamilyCharacterMapInitialized && !TestCharacterMap(aMatchData->mCh)) {
+    if (mFamilyCharacterMapInitialized && !TestCharacterMap(aMatchData->mCh) &&
+        !prefersColor) {
       // none of the faces in the family support the required char,
       // so bail out immediately
       return;
@@ -1654,7 +1656,13 @@ void gfxFontFamily::FindFontForChar(GlobalFontMatch* aMatchData) {
     }
 
     aMatchData->mCmapsTested++;
-    if (e->HasCharacter(aMatchData->mCh)) {
+    bool supportsChar = e->HasCharacter(aMatchData->mCh);
+    if (!supportsChar && prefersColor) {
+      RefPtr<gfxFont> font = e->FindOrMakeFont(&aMatchData->mStyle);
+      supportsChar = font && font->HasColorGlyphFor(aMatchData->mCh,
+                                                     aMatchData->mNextCh);
+    }
+    if (supportsChar) {
       aMatchData->mCount++;
 
       LogModule* log = gfxPlatform::GetLog(eGfxLog_textrun);
