diff --git a/gfx/thebes/gfxPlatform.cpp b/gfx/thebes/gfxPlatform.cpp
index e0b9542a13da..d7921f049d07 100644
--- a/gfx/thebes/gfxPlatform.cpp
+++ b/gfx/thebes/gfxPlatform.cpp
@@ -3359,7 +3359,7 @@ void gfxPlatform::InitWebGPUConfig() {
     featureExternalTexture.Disable(FeatureStatus::Blocklisted, message.get(),
                                    failureId);
   }
-#if !defined(XP_WIN) && !defined(XP_MACOSX)
+#if !defined(XP_WIN) && !defined(XP_MACOSX) && !defined(XP_IOS)
   featureExternalTexture.ForceDisable(
       FeatureStatus::Blocked,
       "WebGPU external textures are not supported on this Operating System",
