diff --git a/gfx/thebes/gfxPlatformMac.cpp b/gfx/thebes/gfxPlatformMac.cpp
index 12d8567ec85a..c4d8234ded39 100644
--- a/gfx/thebes/gfxPlatformMac.cpp
+++ b/gfx/thebes/gfxPlatformMac.cpp
@@ -22,6 +22,8 @@
 #ifdef MOZ_WIDGET_COCOA
 #  include "mozilla/MacAutoreleasePool.h"
 #  include "nsCocoaFeatures.h"
+#else
+#  include "gfxUIKitVsyncSource.h"
 #endif
 #include "nsComponentManagerUtils.h"
 #include "nsIFile.h"
@@ -1036,8 +1038,22 @@ gfxPlatformMac::CreateGlobalHardwareVsyncSource() {
   osxVsyncSource->DisableVsync();
   return osxVsyncSource.forget();
 #else
-  // TODO: CADisplayLink
-  return GetSoftwareVsyncSource();
+  RefPtr<VsyncSource> uikitVsyncSource = CreateUIKitVsyncSource();
+  if (!uikitVsyncSource) {
+    gfxWarning()
+        << "UIKit Vsync source not created. Falling back to software vsync.";
+    return GetSoftwareVsyncSource();
+  }
+
+  uikitVsyncSource->EnableVsync();
+  if (!uikitVsyncSource->IsVsyncEnabled()) {
+    gfxWarning()
+        << "UIKit Vsync source not enabled. Falling back to software vsync.";
+    return GetSoftwareVsyncSource();
+  }
+
+  uikitVsyncSource->DisableVsync();
+  return uikitVsyncSource.forget();
 #endif
 }
 
