diff --git a/gfx/thebes/SharedFontList.cpp b/gfx/thebes/SharedFontList.cpp
index 1f3bd63aa5fc..b87531792bd4 100644
--- a/gfx/thebes/SharedFontList.cpp
+++ b/gfx/thebes/SharedFontList.cpp
@@ -501,6 +501,8 @@ Face* Family::FindFaceForStyle(FontList* aList, const gfxFontStyle& aStyle,
 
 void Family::SearchAllFontsForChar(FontList* aList,
                                    GlobalFontMatch* aMatchData) {
+  const bool prefersColor = PrefersColor(aMatchData->mPresentation);
+
   auto* charmap = mCharacterMap.ToPtr<const SharedBitSet>(aList);
   if (!charmap) {
     // If the face list is not yet initialized, or if character maps have
@@ -515,7 +517,7 @@ void Family::SearchAllFontsForChar(FontList* aList,
     }
     charmap = mCharacterMap.ToPtr<const SharedBitSet>(aList);
   }
-  if (charmap && !charmap->test(aMatchData->mCh)) {
+  if (charmap && !charmap->test(aMatchData->mCh) && !prefersColor) {
     return;
   }
 
@@ -539,7 +541,7 @@ void Family::SearchAllFontsForChar(FontList* aList,
     // Check style distance if the char is supported, or if charmap not known
     // (so that we don't trigger cmap-loading for faces that would be a worse
     // match than what we've already found).
-    if (!charmap || charmap->test(aMatchData->mCh)) {
+    if (!charmap || charmap->test(aMatchData->mCh) || prefersColor) {
       double distance = WSSDistance(face, aMatchData->mStyle);
       if (distance < aMatchData->mMatchDistance) {
         // It's a better style match: get a fontEntry, and if we haven't
@@ -552,8 +554,13 @@ void Family::SearchAllFontsForChar(FontList* aList,
         if (!fe) {
           continue;
         }
-        if (!charmap && !fe->HasCharacter(aMatchData->mCh)) {
-          continue;
+        if ((!charmap || !charmap->test(aMatchData->mCh)) &&
+            !fe->HasCharacter(aMatchData->mCh)) {
+          RefPtr<gfxFont> font = fe->FindOrMakeFont(&aMatchData->mStyle);
+          if (!font ||
+              !font->HasColorGlyphFor(aMatchData->mCh, aMatchData->mNextCh)) {
+            continue;
+          }
         }
         if (aMatchData->mPresentation != FontPresentation::Any) {
           RefPtr<gfxFont> font = fe->FindOrMakeFont(&aMatchData->mStyle);
