diff --git a/dom/media/CubebUtils.cpp b/dom/media/CubebUtils.cpp
index 8ab1d41cdbb4..68a4c0873f02 100644
--- a/dom/media/CubebUtils.cpp
+++ b/dom/media/CubebUtils.cpp
@@ -71,7 +71,8 @@
 #define PREF_AUDIOIPC_STACK_SIZE "media.audioipc.stack_size"
 #define PREF_AUDIOIPC_SHM_AREA_SIZE "media.audioipc.shm_area_size"
 
-#if defined(XP_LINUX) || defined(XP_MACOSX) || defined(XP_WIN)
+#if defined(XP_LINUX) || defined(XP_MACOSX) || defined(XP_WIN) || \
+    defined(XP_IOS)
 #  define MOZ_CUBEB_REMOTING
 #endif
 
@@ -167,6 +168,7 @@ void* sServerHandle = nullptr;
 
 // Initialized during early startup, protected by sMutex.
 StaticAutoPtr<ipc::FileDescriptor> sIPCConnection;
+bool sIPCConnectionRequested = false;
 
 static bool StartAudioIPCServer() {
   if (sCubebSandbox) {
@@ -578,6 +580,14 @@ void InitBrandName() {
 #ifdef MOZ_CUBEB_REMOTING
 void InitAudioIPCConnection() {
   MOZ_ASSERT(NS_IsMainThread());
+  {
+    StaticMutexAutoLock lock(sMutex);
+    if (sIPCConnection || sIPCConnectionRequested) {
+      return;
+    }
+    sIPCConnectionRequested = true;
+  }
+
   auto contentChild = dom::ContentChild::GetSingleton();
   auto promise = contentChild->SendCreateAudioIPCConnection();
   promise->Then(
