diff --git a/dom/media/moz.build b/dom/media/moz.build
index 354bba2e996f..8dcef4c251b3 100644
--- a/dom/media/moz.build
+++ b/dom/media/moz.build
@@ -25,6 +25,9 @@ component_av = ("Core", "WebRTC: Audio/Video")
 with Files("GetUserMedia*"):
     BUG_COMPONENT = component_av
 
+if CONFIG["MOZ_WEBRTC"]:
+    DIRS += ["bridge"]
+
 DIRS += [
     "autoplay",
     "doctor",
@@ -60,9 +63,6 @@ DIRS += [
 if CONFIG["MOZ_ANDROID_HLS_SUPPORT"]:
     DIRS += ["hls"]
 
-if CONFIG["MOZ_WEBRTC"]:
-    DIRS += ["bridge"]
-
 if CONFIG["MOZ_WEBSPEECH"]:
     DIRS += ["webspeech"]
 
