diff --git a/gfx/webrender_bindings/WebRenderAPI.h b/gfx/webrender_bindings/WebRenderAPI.h
index 0bc4dacb9422..b2f04ed17585 100644
--- a/gfx/webrender_bindings/WebRenderAPI.h
+++ b/gfx/webrender_bindings/WebRenderAPI.h
@@ -210,7 +210,8 @@ class TransactionBuilder final {
   void AddRawFont(wr::FontKey aKey, wr::Vec<uint8_t>& aBytes, uint32_t aIndex);
 
   void AddFontDescriptor(wr::FontKey aKey, wr::Vec<uint8_t>& aBytes,
-                         uint32_t aIndex);
+                         uint32_t aIndex, bool aIsSystemFont,
+                         uint32_t aSystemFontTraits);
 
   void DeleteFont(wr::FontKey aKey);
 
