diff --git a/toolkit/library/moz.build b/toolkit/library/moz.build
index a713a301040f..82facaaa67a3 100644
--- a/toolkit/library/moz.build
+++ b/toolkit/library/moz.build
@@ -253,6 +253,8 @@ if CONFIG["MOZ_WIDGET_TOOLKIT"] == "cocoa":
 
 if CONFIG["MOZ_WIDGET_TOOLKIT"] == "uikit":
     OS_LIBS += [
+        "-framework CoreLocation",
+        "-framework CoreMotion",
         "-framework CoreGraphics",
         "-framework CoreText",
         "-framework OpenGLES",
