diff --git a/dom/media/webrtc/jsapi/PeerConnectionImpl.cpp b/dom/media/webrtc/jsapi/PeerConnectionImpl.cpp
index e9afe2cfb271..3d790928567c 100644
--- a/dom/media/webrtc/jsapi/PeerConnectionImpl.cpp
+++ b/dom/media/webrtc/jsapi/PeerConnectionImpl.cpp
@@ -10,7 +10,6 @@
 #include <sstream>
 #include <vector>
 
-#include "IPeerConnection.h"
 #include "MediaTrackGraph.h"
 #include "PeerConnectionCtx.h"
 #include "RTCDataChannelDeclarations.h"
@@ -149,6 +148,15 @@ static mozilla::LazyLogModule logModuleInfo("signaling");
 
 // Getting exceptions back down from PCObserver is generally not harmful.
 namespace {
+constexpr int32_t kActionOffer = 0;
+constexpr int32_t kActionAnswer = 1;
+constexpr int32_t kActionPRAnswer = 2;
+constexpr int32_t kActionRollback = 3;
+
+constexpr uint16_t kDataChannelReliable = 0;
+constexpr uint16_t kDataChannelPartialReliableRexmit = 1;
+constexpr uint16_t kDataChannelPartialReliableTimed = 2;
+
 // This is a terrible hack.  The problem is that SuppressException is not
 // inline, and we link this file without libxul in some cases (e.g. for our test
 // setup).  So we can't use ErrorResult or IgnoredErrorResult because those call
@@ -1012,14 +1020,14 @@ PeerConnectionImpl::CreateDataChannel(
   Nullable<uint16_t> maxLifeTime;
   Nullable<uint16_t> maxRetransmits;
   switch (aType) {
-    case IPeerConnection::kDataChannelReliable:
+    case kDataChannelReliable:
       prPolicy = DataChannelReliabilityPolicy::Reliable;
       break;
-    case IPeerConnection::kDataChannelPartialReliableRexmit:
+    case kDataChannelPartialReliableRexmit:
       prPolicy = DataChannelReliabilityPolicy::LimitedRetransmissions;
       maxRetransmits.SetValue(aMaxNum);
       break;
-    case IPeerConnection::kDataChannelPartialReliableTimed:
+    case kDataChannelPartialReliableTimed:
       prPolicy = DataChannelReliabilityPolicy::LimitedLifetime;
       maxLifeTime.SetValue(aMaxTime);
       break;
@@ -1592,16 +1600,16 @@ PeerConnectionImpl::SetLocalDescription(int32_t aAction, const char* aSDP) {
   bool wasRestartingIce = mJsepSession->IsIceRestarting();
   JsepSdpType sdpType;
   switch (aAction) {
-    case IPeerConnection::kActionOffer:
+    case kActionOffer:
       sdpType = mozilla::kJsepSdpOffer;
       break;
-    case IPeerConnection::kActionAnswer:
+    case kActionAnswer:
       sdpType = mozilla::kJsepSdpAnswer;
       break;
-    case IPeerConnection::kActionPRAnswer:
+    case kActionPRAnswer:
       sdpType = mozilla::kJsepSdpPranswer;
       break;
-    case IPeerConnection::kActionRollback:
+    case kActionRollback:
       sdpType = mozilla::kJsepSdpRollback;
       break;
     default:
@@ -1662,7 +1670,7 @@ PeerConnectionImpl::SetRemoteDescription(int32_t action, const char* aSDP) {
     return NS_ERROR_FAILURE;
   }
 
-  if (action == IPeerConnection::kActionOffer) {
+  if (action == kActionOffer) {
     if (!PeerConnectionCtx::GetInstance()->isReady()) {
       // Uh oh. We're not ready yet. Enqueue this operation. (This must be a
       // remote offer, or else we would not have gotten this far)
@@ -1691,16 +1699,16 @@ PeerConnectionImpl::SetRemoteDescription(int32_t action, const char* aSDP) {
   bool wasRestartingIce = mJsepSession->IsIceRestarting();
   JsepSdpType sdpType;
   switch (action) {
-    case IPeerConnection::kActionOffer:
+    case kActionOffer:
       sdpType = mozilla::kJsepSdpOffer;
       break;
-    case IPeerConnection::kActionAnswer:
+    case kActionAnswer:
       sdpType = mozilla::kJsepSdpAnswer;
       break;
-    case IPeerConnection::kActionPRAnswer:
+    case kActionPRAnswer:
       sdpType = mozilla::kJsepSdpPranswer;
       break;
-    case IPeerConnection::kActionRollback:
+    case kActionRollback:
       sdpType = mozilla::kJsepSdpRollback;
       break;
     default:
