diff --git a/toolkit/modules/ActorManagerParent.sys.mjs b/toolkit/modules/ActorManagerParent.sys.mjs
index f71c43c6dafb..cf374e9be166 100644
--- a/toolkit/modules/ActorManagerParent.sys.mjs
+++ b/toolkit/modules/ActorManagerParent.sys.mjs
@@ -713,7 +713,9 @@ if (AppConstants.platform != "android") {
     allFrames: true,
     safeForUntrustedWebProcess: true,
   };
+}
 
+if (AppConstants.platform != "android" && AppConstants.platform != "ios") {
   // Note that GeckoView has another implementation in mobile/android/actors.
   JSWINDOWACTORS.Select = {
     parent: {
