diff --git a/layout/base/PresShell.h b/layout/base/PresShell.h
index 23efeb53f7c7..6ebb640ceb2c 100644
--- a/layout/base/PresShell.h
+++ b/layout/base/PresShell.h
@@ -1646,6 +1646,11 @@ class PresShell final : public nsStubDocumentObserver,
 
   nsPoint GetVisualViewportOffsetRelativeToLayoutViewport() const;
 
+  void SetFixedPositionMargins(const Maybe<CSSMargin>& aMargins);
+  const Maybe<CSSMargin>& GetFixedPositionMargins() const {
+    return mFixedPositionMargins;
+  }
+
   // Returns state of the dynamic toolbar.
   DynamicToolbarState GetDynamicToolbarState() const {
     if (!mPresContext) {
@@ -1658,8 +1663,7 @@ class PresShell final : public nsStubDocumentObserver,
   // shown/hidden.
   nsSize GetVisualViewportSizeUpdatedByDynamicToolbar() const;
 
-  // Returns the fixed viewport size accounted for
-  // a fully shown or fully hidden dynamic toolbar
+  // Returns the fixed viewport size accounting for the current toolbar height.
   nsSize GetFixedViewportSize() const;
 
   // Trigger refreshing the MobileViewportManager's size metrics.
@@ -3416,6 +3420,9 @@ class PresShell final : public nsStubDocumentObserver,
 
   Maybe<nsPoint> mVisualViewportOffset;
 
+  // Compositor margins relative to the last content paint, in CSS pixels.
+  Maybe<CSSMargin> mFixedPositionMargins;
+
   // A pending visual scroll offset that we will ask APZ to scroll to
   // during the next transaction. Cleared when we send the transaction.
   // Only applicable to the RCD pres shell.
