diff --git a/layout/base/PresShell.cpp b/layout/base/PresShell.cpp
index 8b1f69b52380..646031091357 100644
--- a/layout/base/PresShell.cpp
+++ b/layout/base/PresShell.cpp
@@ -47,6 +47,7 @@
 #include "mozilla/IntegerRange.h"
 #include "mozilla/Likely.h"
 #include "mozilla/Logging.h"
+#include "mozilla/LookAndFeel.h"
 #include "mozilla/MemoryReporting.h"
 #include "mozilla/MouseEvents.h"
 #include "mozilla/PerfStats.h"
@@ -5771,7 +5772,32 @@ nscolor PresShell::GetDefaultBackgroundColorToDraw() const {
 }
 
 void PresShell::UpdateCanvasBackground() {
-  mCanvasBackground = ComputeCanvasBackground();
+  CanvasBackground canvasBackground = ComputeCanvasBackground();
+
+#ifdef XP_IOS
+  const auto& oldViewportBackground = mCanvasBackground.mViewport;
+  const auto& newViewportBackground = canvasBackground.mViewport;
+  const bool backgroundChanged =
+      oldViewportBackground.mColor != newViewportBackground.mColor ||
+      oldViewportBackground.mCSSSpecified !=
+          newViewportBackground.mCSSSpecified;
+  if (backgroundChanged && this == GetRootPresShell() &&
+      mDocument->GetBrowsingContext() &&
+      mDocument->GetBrowsingContext()->IsTopContent()) {
+    nscolor backgroundColor = newViewportBackground.mColor;
+    if (!newViewportBackground.mCSSSpecified) {
+      backgroundColor = LookAndFeel::Color(
+          LookAndFeel::ColorID::Window, LookAndFeel::SystemColorScheme(),
+          LookAndFeel::UseStandins::No, NS_RGB(255, 255, 255));
+    }
+    if (BrowserChild* browserChild =
+            BrowserChild::GetFrom(mDocument->GetDocShell())) {
+      browserChild->UpdateBackgroundColor(backgroundColor);
+    }
+  }
+#endif
+
+  mCanvasBackground = canvasBackground;
 }
 
 static SingleCanvasBackground ComputeSingleCanvasBackground(nsIFrame* aCanvas) {
@@ -12471,6 +12497,17 @@ nsPoint PresShell::GetVisualViewportOffsetRelativeToLayoutViewport() const {
   return GetVisualViewportOffset() - GetLayoutViewportOffset();
 }
 
+void PresShell::SetFixedPositionMargins(const Maybe<CSSMargin>& aMargins) {
+  if (mFixedPositionMargins == aMargins) {
+    return;
+  }
+  mFixedPositionMargins = aMargins;
+  if (nsIWidget* widget = GetOwnWidget();
+      widget && widget->GetInputContext().mIMEState.IsEditable()) {
+    IMEStateManager::NotifyIME(widget::NOTIFY_IME_OF_POSITION_CHANGE, widget);
+  }
+}
+
 nsPoint PresShell::GetLayoutViewportOffset() const {
   nsPoint result;
   if (ScrollContainerFrame* sf = GetRootScrollContainerFrame()) {
@@ -12516,7 +12553,7 @@ nsSize PresShell::GetFixedViewportSize() const {
     return layoutViewportSize;
   }
   layoutViewportSize.height +=
-      mPresContext->GetBimodalDynamicToolbarHeightForFixedPosInAppUnits();
+      mPresContext->GetHiddenDynamicToolbarHeightForFixedPosInAppUnits();
   return layoutViewportSize;
 }
 
