diff --git a/dom/media/systemservices/CamerasParent.h b/dom/media/systemservices/CamerasParent.h
index 18fe4aba162e..3bbc7cb306c2 100644
--- a/dom/media/systemservices/CamerasParent.h
+++ b/dom/media/systemservices/CamerasParent.h
@@ -180,7 +180,8 @@ class CamerasParent : public PCamerasParent {
    *   The returned promise will never be rejected.
    */
   static RefPtr<CameraAccessRequestPromise> RequestCameraAccess(
-      bool aAllowPermissionRequest);
+      bool aAllowPermissionRequest, bool aRequestCamera,
+      bool aRequestMicrophone);
 
   // Messages received from the child. These run on the IPC/PBackground thread.
   mozilla::ipc::IPCResult RecvPCamerasConstructor();
