diff --git a/dom/media/systemservices/CamerasParent.h b/dom/media/systemservices/CamerasParent.h
index 44556c7158f2..5ca849a2be62 100644
--- a/dom/media/systemservices/CamerasParent.h
+++ b/dom/media/systemservices/CamerasParent.h
@@ -176,7 +176,8 @@ class CamerasParent : public PCamerasParent {
    *   The returned promise will never be rejected.
    */
   static RefPtr<CameraAccessRequestPromise> RequestCameraAccess(
-      bool aAllowPermissionRequest);
+      bool aAllowPermissionRequest, bool aRequestCamera,
+      bool aRequestMicrophone);
 
   // Messages received from the child. These run on the IPC/PBackground thread.
   mozilla::ipc::IPCResult RecvPCamerasConstructor();
