diff --git a/dom/media/systemservices/PCameras.ipdl b/dom/media/systemservices/PCameras.ipdl
index c9652718acde..95d3a219749d 100644
--- a/dom/media/systemservices/PCameras.ipdl
+++ b/dom/media/systemservices/PCameras.ipdl
@@ -8,12 +8,18 @@ include protocol PBackground;
 include PBackgroundSharedTypes;
 
 include "mozilla/dom/MediaIPCUtils.h";
+include "mozilla/GfxMessageUtils.h";
 
 using mozilla::camera::CaptureEngine from "mozilla/media/CamerasTypes.h";
 using mozilla::dom::VideoResizeModeEnum from "mozilla/dom/MediaStreamTrackBinding.h";
 using mozilla::TimeStamp from "mozilla/TimeStamp.h";
 using mozilla::media::TimeUnit from "TimeUnits.h";
 using mozilla::NormalizedConstraints from "MediaTrackConstraints.h";
+using mozilla::gfx::IntRect from "mozilla/gfx/Rect.h";
+using mozilla::gfx::YUVColorSpace from "mozilla/gfx/Types.h";
+using mozilla::gfx::ColorSpace2 from "mozilla/gfx/Types.h";
+using mozilla::gfx::TransferFunction from "mozilla/gfx/Types.h";
+[MoveOnly] using mozilla::layers::IOSurfacePort from "mozilla/layers/IOSurfacePort.h";
 using webrtc::VideoType from "mozilla/media/CamerasTypes.h";
 
 namespace mozilla {
@@ -24,6 +30,7 @@ struct VideoCaptureCapability
 {
   int width;
   int height;
+  int minFPS;
   int maxFPS;
   VideoType videoType;
   bool interlaced;
@@ -65,12 +72,19 @@ child:
   async CaptureEnded(int[] streamIds);
   // transfers ownership of |buffer| from parent to child
   async DeliverFrame(int captureId, int[] streamIds, Shmem buffer, VideoFrameProperties props);
+  async DeliverFrameIOSurface(int captureId, uint64_t frameId, int[] streamIds,
+                              IOSurfacePort surface, IntRect pictureRect,
+                              YUVColorSpace yUVColorSpace,
+                              ColorSpace2 colorPrimaries,
+                              TransferFunction transferFunction,
+                              VideoFrameProperties props);
   async DeviceChange();
   async ReplyNumberOfCaptureDevices(int deviceCount);
   async ReplyNumberOfCapabilities(int capabilityCount);
   async ReplyAllocateCapture(int streamId);
   async ReplyGetCaptureCapability(VideoCaptureCapability cap);
-  async ReplyGetCaptureDevice(nsCString device_name, nsCString device_id, bool scary);
+  async ReplyGetCaptureDevice(nsCString device_name, nsCString device_id,
+                              bool scary, uint8_t facingMode);
   async ReplyFailure();
   async ReplySuccess();
   async __delete__();
@@ -95,6 +109,7 @@ parent:
   async StopCapture(CaptureEngine engine, int streamId);
   // transfers frame back
   async ReleaseFrame(int captureId, Shmem s);
+  async ReleaseFrameIOSurface(int captureId, uint64_t frameId);
 
   // setup camera engine
   async EnsureInitialized(CaptureEngine engine);
