diff --git a/dom/media/platforms/apple/AppleVTDecoder.cpp b/dom/media/platforms/apple/AppleVTDecoder.cpp
index 6881412889f5..b4f3d750d173 100644
--- a/dom/media/platforms/apple/AppleVTDecoder.cpp
+++ b/dom/media/platforms/apple/AppleVTDecoder.cpp
@@ -6,6 +6,9 @@
 
 #include <CoreVideo/CVPixelBufferIOSurface.h>
 #include <IOSurface/IOSurfaceRef.h>
+#ifdef MOZ_WIDGET_UIKIT
+#  include <TargetConditionals.h>
+#endif
 
 #include <limits>
 
@@ -65,7 +68,13 @@ AppleVTDecoder::AppleVTDecoder(const VideoInfo& aConfig,
       mKnowsCompositor(aKnowsCompositor)
 #ifdef MOZ_WIDGET_UIKIT
       ,
+#  if TARGET_OS_SIMULATOR
       mUseSoftwareImages(true)
+#  else
+      mUseSoftwareImages(!aKnowsCompositor ||
+                         aKnowsCompositor->GetWebRenderCompositorType() ==
+                             layers::WebRenderCompositor::SOFTWARE)
+#  endif
 #else
       ,
       mUseSoftwareImages(aKnowsCompositor &&
@@ -435,7 +444,7 @@ nsCString AppleVTDecoder::GetCodecName() const {
 
 CFStringRef CGColorSpaceNameForFrame(gfx::TransferFunction aTransferFunction,
                                      gfx::ColorSpace2 aColorPrimaries) {
-  if (__builtin_available(macOS 11.0, *)) {
+  if (__builtin_available(macOS 11.0, iOS 14.0, *)) {
     if (aTransferFunction == gfx::TransferFunction::PQ) {
       return kCGColorSpaceITUR_2100_PQ;
     }
@@ -763,22 +772,24 @@ MediaResult AppleVTDecoder::InitializeSession() {
                        RESULT_DETAIL("Couldn't create decompression session!"));
   }
 
-  CFBooleanRef isUsingHW = nullptr;
-  rv = VTSessionCopyProperty(
-      mSession,
-      kVTDecompressionPropertyKey_UsingHardwareAcceleratedVideoDecoder,
-      kCFAllocatorDefault, &isUsingHW);
-  if (rv == noErr) {
-    mIsHardwareAccelerated = isUsingHW == kCFBooleanTrue;
-    LOG("AppleVTDecoder: {} hardware accelerated decoding",
-        mIsHardwareAccelerated ? "using" : "not using");
-  } else {
-    LOG("AppleVTDecoder: maybe hardware accelerated decoding "
-        "(VTSessionCopyProperty query failed {})",
-        static_cast<int>(rv));
-  }
-  if (isUsingHW) {
-    CFRelease(isUsingHW);
+  if (__builtin_available(iOS 17.0, *)) {
+    CFBooleanRef isUsingHW = nullptr;
+    rv = VTSessionCopyProperty(
+        mSession,
+        kVTDecompressionPropertyKey_UsingHardwareAcceleratedVideoDecoder,
+        kCFAllocatorDefault, &isUsingHW);
+    if (rv == noErr) {
+      mIsHardwareAccelerated = isUsingHW == kCFBooleanTrue;
+      LOG("AppleVTDecoder: {} hardware accelerated decoding",
+          mIsHardwareAccelerated ? "using" : "not using");
+    } else {
+      LOG("AppleVTDecoder: maybe hardware accelerated decoding "
+          "(VTSessionCopyProperty query failed {})",
+          static_cast<int>(rv));
+    }
+    if (isUsingHW) {
+      CFRelease(isUsingHW);
+    }
   }
 
   return NS_OK;
@@ -825,6 +836,10 @@ CFDictionaryRef AppleVTDecoder::CreateDecoderExtensions() {
 }
 
 CFDictionaryRef AppleVTDecoder::CreateDecoderSpecification() {
+  if (!__builtin_available(iOS 17.0, *)) {
+    return nullptr;
+  }
+
   const void* specKeys[] = {
       kVTVideoDecoderSpecification_EnableHardwareAcceleratedVideoDecoder};
   const void* specValues[1];
@@ -881,9 +896,15 @@ CFDictionaryRef AppleVTDecoder::CreateOutputConfiguration() {
       std::size(IOSurfaceKeys), &kCFTypeDictionaryKeyCallBacks,
       &kCFTypeDictionaryValueCallBacks));
 
+  const void* compatibilityKey =
+#ifdef MOZ_WIDGET_UIKIT
+      kCVPixelBufferOpenGLESCompatibilityKey;
+#else
+      kCVPixelBufferOpenGLCompatibilityKey;
+#endif
   const void* outputKeys[] = {kCVPixelBufferIOSurfacePropertiesKey,
                               kCVPixelBufferPixelFormatTypeKey,
-                              kCVPixelBufferOpenGLCompatibilityKey};
+                              compatibilityKey};
   const void* outputValues[] = {IOSurfaceProperties, PixelFormatTypeNumber,
                                 kCFBooleanTrue};
   static_assert(std::size(outputKeys) == std::size(outputValues),
