diff --git a/gfx/layers/wr/WebRenderBridgeParent.cpp b/gfx/layers/wr/WebRenderBridgeParent.cpp
index a58c4d319439..bec31ca2ccbb 100644
--- a/gfx/layers/wr/WebRenderBridgeParent.cpp
+++ b/gfx/layers/wr/WebRenderBridgeParent.cpp
@@ -776,7 +776,8 @@ bool WebRenderBridgeParent::UpdateResources(
 
         wr::Vec<uint8_t> bytes;
         if (reader.Read(op.bytes(), bytes)) {
-          aUpdates.AddFontDescriptor(op.key(), bytes, op.fontIndex());
+          aUpdates.AddFontDescriptor(op.key(), bytes, op.fontIndex(),
+                                     op.isSystemFont(), op.systemFontTraits());
         } else {
           gfxCriticalNote << "TOpAddFontDescriptor failed";
           success = false;
