diff --git a/gfx/layers/wr/WebRenderBridgeChild.cpp b/gfx/layers/wr/WebRenderBridgeChild.cpp
index 036f1a9bed0c..9f1ae81d8048 100644
--- a/gfx/layers/wr/WebRenderBridgeChild.cpp
+++ b/gfx/layers/wr/WebRenderBridgeChild.cpp
@@ -221,6 +221,8 @@ struct FontFileDataSink {
   wr::FontKey* mFontKey;
   WebRenderBridgeChild* mWrBridge;
   wr::IpcResourceUpdateQueue* mResources;
+  bool mIsSystemFont;
+  uint32_t mSystemFontTraits;
 };
 
 static void WriteFontFileData(const uint8_t* aData, uint32_t aLength,
@@ -242,7 +244,7 @@ static void WriteFontDescriptor(const uint8_t* aData, uint32_t aLength,
 
   sink->mResources->AddFontDescriptor(
       *sink->mFontKey, Range<uint8_t>(const_cast<uint8_t*>(aData), aLength),
-      aIndex);
+      aIndex, sink->mIsSystemFont, sink->mSystemFontTraits);
 }
 
 void WebRenderBridgeChild::PushGlyphs(
@@ -305,7 +307,11 @@ Maybe<wr::FontKey> WebRenderBridgeChild::GetFontKeyForUnscaledFont(
       aUnscaled, [&](auto&& entry) -> Maybe<wr::FontKey> {
         if (!entry) {
           wr::FontKey fontKey = {wr::IdNamespace{0}, 0};
-          FontFileDataSink sink = {&fontKey, this, &aResources};
+          FontFileDataSink sink = {
+              &fontKey, this, &aResources,
+              aUnscaled->GetDescriptorProvenance() ==
+                  gfx::UnscaledFont::DescriptorProvenance::SystemUI,
+              aUnscaled->GetSystemFontTraits()};
           // First try to retrieve a descriptor for the font, as this is much
           // cheaper to send over IPC than the full raw font data. If this is
           // not possible, then and only then fall back to getting the raw font
