diff --git a/dom/geolocation/Geolocation.cpp b/dom/geolocation/Geolocation.cpp
index 51efcf97c062..671c52010401 100644
--- a/dom/geolocation/Geolocation.cpp
+++ b/dom/geolocation/Geolocation.cpp
@@ -291,6 +291,12 @@ NS_IMPL_ISUPPORTS0(CancelSystemGeolocationPermissionRequest)
 void Geolocation::ReallowWithSystemPermissionOrCancel(
     BrowsingContext* aBrowsingContext,
     geolocation::ParentRequestResolver&& aResolver) {
+#ifdef MOZ_WIDGET_UIKIT
+  RefPtr<SystemGeolocationPermissionRequest> permissionRequest =
+      geolocation::RequestLocationPermissionFromUser(aBrowsingContext,
+                                                     std::move(aResolver));
+  return;
+#else
   // Make sure we don't return without responding to the geolocation request.
   auto denyPermissionOnError =
       MakeScopeExit([&aResolver]() MOZ_CAN_RUN_SCRIPT_BOUNDARY_LAMBDA {
@@ -372,6 +378,7 @@ void Geolocation::ReallowWithSystemPermissionOrCancel(
       new CancelSystemGeolocationPermissionRequest(permissionRequest));
 
   cancelRequestOnError.release();
+#endif
 }
 
 NS_IMETHODIMP
