diff --git a/mobile/shared/modules/geckoview/moz.build b/mobile/shared/modules/geckoview/moz.build
index b0a38a3b876f..0cebdeffc761 100644
--- a/mobile/shared/modules/geckoview/moz.build
+++ b/mobile/shared/modules/geckoview/moz.build
@@ -6,6 +6,8 @@ EXTRA_JS_MODULES += [
     "BrowserUsageTelemetry.sys.mjs",
     "ChildCrashHandler.sys.mjs",
     "DelayedInit.sys.mjs",
+    "FormAutoFillKeywords.sys.mjs",
+    "FormFieldClassifier.sys.mjs",
     "GeckoViewActorChild.sys.mjs",
     "GeckoViewActorManager.sys.mjs",
     "GeckoViewActorParent.sys.mjs",
