diff --git a/netwerk/ipc/SocketProcessHost.cpp b/netwerk/ipc/SocketProcessHost.cpp
index 710de656a8e4..98763d54e613 100644
--- a/netwerk/ipc/SocketProcessHost.cpp
+++ b/netwerk/ipc/SocketProcessHost.cpp
@@ -72,7 +72,7 @@ bool SocketProcessHost::Launch() {
   ProcessChild::AddPlatformBuildID(extraArgs);
 
   SharedPreferenceSerializer prefSerializer;
-  if (!prefSerializer.SerializeToSharedMemory(GeckoProcessType_VR,
+  if (!prefSerializer.SerializeToSharedMemory(GeckoProcessType_Socket,
                                               /* remoteType */ {})) {
     return false;
   }
