diff --git a/netwerk/protocol/http/HttpConnectionMgrParent.h b/netwerk/protocol/http/HttpConnectionMgrParent.h
index c264f8864e48..e25a3ef45a95 100644
--- a/netwerk/protocol/http/HttpConnectionMgrParent.h
+++ b/netwerk/protocol/http/HttpConnectionMgrParent.h
@@ -7,8 +7,11 @@
 
 #include "HttpConnectionMgrShell.h"
 #include "mozilla/StaticMutex.h"
+#include "mozilla/ipc/PBackgroundSharedTypes.h"
 #include "mozilla/net/PHttpConnectionMgrParent.h"
 
+class nsIWebTransportHash;
+
 namespace mozilla::net {
 
 // HttpConnectionMgrParent plays the role of nsHttpConnectionMgr and delegates
@@ -21,6 +24,10 @@ class HttpConnectionMgrParent final : public PHttpConnectionMgrParent,
 
   explicit HttpConnectionMgrParent() = default;
 
+  [[nodiscard]] nsresult StoreServerCertHashes(
+      nsHttpConnectionInfo* aConnInfo, bool aNoSpdy, bool aNoHttp3,
+      nsTArray<RefPtr<nsIWebTransportHash>>&& aServerCertHashes);
+
   static uint32_t AddHttpUpgradeListenerToMap(
       nsIHttpUpgradeListener* aListener);
   static void RemoveHttpUpgradeListenerFromMap(uint32_t aId);
