diff --git a/netwerk/protocol/http/nsHttpHandler.cpp b/netwerk/protocol/http/nsHttpHandler.cpp
index c45cb9f04196..f9f91718ef1a 100644
--- a/netwerk/protocol/http/nsHttpHandler.cpp
+++ b/netwerk/protocol/http/nsHttpHandler.cpp
@@ -660,6 +660,23 @@ nsresult nsHttpHandler::InitConnectionMgr() {
                         mBeConservativeForProxy);
 }
 
+nsresult nsHttpHandler::StoreServerCertHashes(
+    nsHttpConnectionInfo* aConnInfo, bool aNoSpdy, bool aNoHttp3,
+    nsTArray<RefPtr<nsIWebTransportHash>>&& aServerCertHashes) {
+  if (nsHttpConnectionMgr* connMgr = ConnMgr()) {
+    return connMgr->StoreServerCertHashes(aConnInfo, aNoSpdy, aNoHttp3,
+                                          std::move(aServerCertHashes));
+  }
+
+  if (HttpConnectionMgrParent* connMgrParent =
+          mConnMgr ? mConnMgr->AsHttpConnectionMgrParent() : nullptr) {
+    return connMgrParent->StoreServerCertHashes(aConnInfo, aNoSpdy, aNoHttp3,
+                                                std::move(aServerCertHashes));
+  }
+
+  return NS_ERROR_NOT_AVAILABLE;
+}
+
 // We're using RequestOverride because this can get called when these are
 // set by Fetch from the old request. We need to pass a function pointer to
 // let GetDictionaryFor suspend the channel before starting the async
