diff --git a/netwerk/protocol/http/nsHttpChannel.cpp b/netwerk/protocol/http/nsHttpChannel.cpp
index c6d926448e39..d51f3e24af66 100644
--- a/netwerk/protocol/http/nsHttpChannel.cpp
+++ b/netwerk/protocol/http/nsHttpChannel.cpp
@@ -1248,9 +1248,10 @@ nsresult nsHttpChannel::ContinueOnBeforeConnect(bool aShouldUpgrade,
     NS_ENSURE_SUCCESS(rv, rv);
 
     wtconSettings->GetServerCertificateHashes(aServerCertHashes);
-    gHttpHandler->ConnMgr()->StoreServerCertHashes(
+    DebugOnly<nsresult> storeRv = gHttpHandler->StoreServerCertHashes(
         mConnectionInfo, gHttpHandler->IsHttp2Excluded(mConnectionInfo),
         !Http3Allowed(), std::move(aServerCertHashes));
+    MOZ_ASSERT(NS_SUCCEEDED(storeRv));
   }
 
   if (ShouldIntercept()) {
