diff --git a/netwerk/protocol/http/nsHttpChannel.cpp b/netwerk/protocol/http/nsHttpChannel.cpp
index 6d6381581bab..a67469f7bc4b 100644
--- a/netwerk/protocol/http/nsHttpChannel.cpp
+++ b/netwerk/protocol/http/nsHttpChannel.cpp
@@ -1215,9 +1215,10 @@ nsresult nsHttpChannel::ContinueOnBeforeConnect(bool aShouldUpgrade,
     NS_ENSURE_SUCCESS(rv, rv);
 
     wtconSettings->GetServerCertificateHashes(aServerCertHashes);
-    gHttpHandler->ConnMgr()->StoreServerCertHashes(
+    DebugOnly<nsresult> storeRv = gHttpHandler->StoreServerCertHashes(
         mConnectionInfo, gHttpHandler->IsHttp2Excluded(mConnectionInfo),
         !Http3Allowed(), std::move(aServerCertHashes));
+    MOZ_ASSERT(NS_SUCCEEDED(storeRv));
   }
 
   if (ShouldIntercept()) {
