diff --git a/netwerk/protocol/http/nsHttpHandler.h b/netwerk/protocol/http/nsHttpHandler.h
index d94db56bde47..b0721065acd9 100644
--- a/netwerk/protocol/http/nsHttpHandler.h
+++ b/netwerk/protocol/http/nsHttpHandler.h
@@ -51,6 +51,7 @@ class nsIRequestContextService;
 class nsISiteIntegrityService;
 class nsISiteSecurityService;
 class nsIStreamConverterService;
+class nsIWebTransportHash;
 
 namespace mozilla::net {
 
@@ -268,6 +269,10 @@ class nsHttpHandler final : public nsIHttpProtocolHandler,
   [[nodiscard]] nsresult InitiateTransaction(HttpTransactionShell* trans,
                                              int32_t priority);
 
+  [[nodiscard]] nsresult StoreServerCertHashes(
+      nsHttpConnectionInfo* aConnInfo, bool aNoSpdy, bool aNoHttp3,
+      nsTArray<RefPtr<nsIWebTransportHash>>&& aServerCertHashes);
+
   // This function is also called to kick-off a new transaction. But the new
   // transaction will take a sticky connection from |transWithStickyConn|
   // and reuse it.
