diff --git a/netwerk/protocol/http/PHttpConnectionMgr.ipdl b/netwerk/protocol/http/PHttpConnectionMgr.ipdl
index 49f56e147351..9236883e0101 100644
--- a/netwerk/protocol/http/PHttpConnectionMgr.ipdl
+++ b/netwerk/protocol/http/PHttpConnectionMgr.ipdl
@@ -7,6 +7,7 @@ include protocol PSocketProcess;
 include protocol PHttpTransaction;
 
 include NeckoChannelParams;
+include PBackgroundSharedTypes;
 
 include "mozilla/net/ClassOfService.h";
 include "mozilla/net/NeckoMessageUtils.h";
@@ -32,6 +33,9 @@ child:
   async UpdateClassOfServiceOnTransaction(PHttpTransaction aTrans,
                                           ClassOfService aClassOfService);
   async CancelTransaction(PHttpTransaction aTrans, nsresult aReason);
+  async StoreServerCertHashes(HttpConnectionInfoCloneArgs aConnInfo,
+                              bool aNoSpdy, bool aNoHttp3,
+                              WebTransportHash[] aServerCertHashes);
   async SpeculativeConnect(HttpConnectionInfoCloneArgs aConnInfo,
                            SpeculativeConnectionOverriderArgs? aOverriderArgs,
                            uint32_t aCaps, PAltSvcTransaction? aTrans,
