diff --git a/ipc/glue/BackgroundParentImpl.cpp b/ipc/glue/BackgroundParentImpl.cpp
index 8007cd47341b..f53b2e08e9d5 100644
--- a/ipc/glue/BackgroundParentImpl.cpp
+++ b/ipc/glue/BackgroundParentImpl.cpp
@@ -1455,10 +1455,11 @@ BackgroundParentImpl::RecvEnsureUtilityProcessAndCreateBridge(
 }
 
 mozilla::ipc::IPCResult BackgroundParentImpl::RecvRequestCameraAccess(
-    const bool& aAllowPermissionRequest,
-    RequestCameraAccessResolver&& aResolver) {
+    const bool& aAllowPermissionRequest, const bool& aRequestCamera,
+    const bool& aRequestMicrophone, RequestCameraAccessResolver&& aResolver) {
 #ifdef MOZ_WEBRTC
-  mozilla::camera::CamerasParent::RequestCameraAccess(aAllowPermissionRequest)
+  mozilla::camera::CamerasParent::RequestCameraAccess(
+      aAllowPermissionRequest, aRequestCamera, aRequestMicrophone)
       ->Then(GetCurrentSerialEventTarget(), __func__,
              [resolver = std::move(aResolver)](
                  const mozilla::camera::CamerasParent::
