diff --git a/ipc/glue/BackgroundParentImpl.h b/ipc/glue/BackgroundParentImpl.h
index 663721274f8f..cccaa851ff70 100644
--- a/ipc/glue/BackgroundParentImpl.h
+++ b/ipc/glue/BackgroundParentImpl.h
@@ -338,7 +338,8 @@ class BackgroundParentImpl : public PBackgroundParent {
       EnsureUtilityProcessAndCreateBridgeResolver&& aResolver) override;
 
   mozilla::ipc::IPCResult RecvRequestCameraAccess(
-      const bool& aAllowPermissionRequest,
+      const bool& aAllowPermissionRequest, const bool& aRequestCamera,
+      const bool& aRequestMicrophone,
       RequestCameraAccessResolver&& aResolver) override;
 
   mozilla::ipc::IPCResult RecvPLockManagerConstructor(
