diff --git a/ipc/glue/BackgroundParentImpl.h b/ipc/glue/BackgroundParentImpl.h
index 429fd9b3f571..7ac573bbef18 100644
--- a/ipc/glue/BackgroundParentImpl.h
+++ b/ipc/glue/BackgroundParentImpl.h
@@ -335,7 +335,8 @@ class BackgroundParentImpl : public PBackgroundParent {
       EnsureUtilityProcessAndCreateBridgeResolver&& aResolver) override;
 
   mozilla::ipc::IPCResult RecvRequestCameraAccess(
-      const bool& aAllowPermissionRequest,
+      const bool& aAllowPermissionRequest, const bool& aRequestCamera,
+      const bool& aRequestMicrophone,
       RequestCameraAccessResolver&& aResolver) override;
 
   mozilla::ipc::IPCResult RecvPLockManagerConstructor(
