diff --git a/ipc/glue/GeckoChildProcessHost.h b/ipc/glue/GeckoChildProcessHost.h
index d3e8b775421b..a21b0d0e9beb 100644
--- a/ipc/glue/GeckoChildProcessHost.h
+++ b/ipc/glue/GeckoChildProcessHost.h
@@ -33,7 +33,7 @@
 #include "nsString.h"
 
 #if defined(XP_IOS)
-#  include "mozilla/ipc/ExtensionKitUtils.h"
+#  include "mozilla/ipc/NSExtensionUtils.h"
 #endif
 
 #if defined(XP_WIN) && defined(MOZ_SANDBOX)
@@ -170,6 +170,10 @@ class GeckoChildProcessHost : public SupportsWeakPtr,
   // For bug 943174: Skip the EnsureProcessTerminated call in the destructor.
   void SetAlreadyDead();
 
+#if defined(MOZ_WIDGET_UIKIT)
+  void ForceInvalidateForTermination();
+#endif
+
 #if defined(MOZ_SANDBOX) && defined(XP_MACOSX)
   // Start the sandbox from the child process.
   static bool StartMacSandbox(int aArgc, char** aArgv,
@@ -268,7 +272,7 @@ class GeckoChildProcessHost : public SupportsWeakPtr,
   task_t mChildTask MOZ_GUARDED_BY(mHandleLock);
 #endif
 #if defined(MOZ_WIDGET_UIKIT)
-  Maybe<ExtensionKitProcess> mExtensionKitProcess MOZ_GUARDED_BY(mHandleLock);
+  Maybe<NSExtensionProcess> mNSExtensionProcess MOZ_GUARDED_BY(mHandleLock);
   DarwinObjectPtr<xpc_connection_t> mXPCConnection MOZ_GUARDED_BY(mHandleLock);
   UniqueBEProcessCapabilityGrant mForegroundCapabilityGrant
       MOZ_GUARDED_BY(mHandleLock);
