diff --git a/ipc/glue/moz.build b/ipc/glue/moz.build
index bfb11d852776..52721665be71 100644
--- a/ipc/glue/moz.build
+++ b/ipc/glue/moz.build
@@ -148,10 +148,10 @@ if CONFIG["OS_ARCH"] != "WINNT":
 
 if CONFIG["TARGET_OS"] == "iOS":
     EXPORTS.mozilla.ipc += [
-        "ExtensionKitUtils.h",
+        "NSExtensionUtils.h",
     ]
     UNIFIED_SOURCES += [
-        "ExtensionKitUtils.mm",
+        "NSExtensionUtils.mm",
     ]
 
 EXPORTS.ipc += [
@@ -288,8 +288,6 @@ FINAL_LIBRARY = "xul"
 
 if CONFIG["TARGET_OS"] == "OSX":
     OS_LIBS += ["bsm"]  # for audit_token_to_pid
-elif CONFIG["TARGET_OS"] == "iOS":
-    OS_LIBS += ["-framework BrowserEngineKit"]
 elif CONFIG["MOZ_WIDGET_TOOLKIT"] == "android":
     OS_LIBS += ["android"]
 
