diff --git a/ipc/glue/PBackground.ipdl b/ipc/glue/PBackground.ipdl
index bfc523c53e07..95142bbe7913 100644
--- a/ipc/glue/PBackground.ipdl
+++ b/ipc/glue/PBackground.ipdl
@@ -296,7 +296,9 @@ parent:
 
   async PFetch();
 
-  async RequestCameraAccess(bool aAllowPermissionRequest) returns (CamerasAccessStatus rv);
+  async RequestCameraAccess(bool aAllowPermissionRequest, bool aRequestCamera,
+                            bool aRequestMicrophone)
+      returns (CamerasAccessStatus rv);
 
 child:
   async PCache();
