diff --git a/modules/woff2/moz.build b/modules/woff2/moz.build
index 661cb8744853..eebf5e3bb64b 100644
--- a/modules/woff2/moz.build
+++ b/modules/woff2/moz.build
@@ -7,7 +7,8 @@ include("sources.mozbuild")
 with Files('**'):
     BUG_COMPONENT = ('Core', 'Graphics: Text')
 
-if not CONFIG["MOZ_WASM_SANDBOXING_WOFF2"]:
+if not CONFIG["MOZ_WASM_SANDBOXING_WOFF2"] or \
+    CONFIG["MOZ_WIDGET_TOOLKIT"] == "uikit":
     UNIFIED_SOURCES += woff2_sources
 
 EXPORTS.woff2 += [
