diff --git a/dom/ipc/BrowserChild.cpp b/dom/ipc/BrowserChild.cpp
index 58037fdbb308..786cc6066e64 100644
--- a/dom/ipc/BrowserChild.cpp
+++ b/dom/ipc/BrowserChild.cpp
@@ -230,6 +230,13 @@ PresShell* BrowserChild::GetTopLevelPresShell() const {
   return nullptr;
 }
 
+void BrowserChild::UpdateBackgroundColor(const nscolor& aColor) {
+  if (!IPCOpen() || mDestroyed) {
+    return;
+  }
+  SendUpdateBackgroundColor(aColor);
+}
+
 bool BrowserChild::UpdateFrame(const RepaintRequest& aRequest) {
   MOZ_ASSERT(aRequest.GetScrollId() != ScrollableLayerGuid::NULL_SCROLL_ID);
 
