diff --git a/dom/ipc/BrowserParent.h b/dom/ipc/BrowserParent.h
index 9f65b8537e4e..06d621fb5cca 100644
--- a/dom/ipc/BrowserParent.h
+++ b/dom/ipc/BrowserParent.h
@@ -389,6 +389,7 @@ class BrowserParent final : public PBrowserParent,
       const nsCursor& aValue, Maybe<IPCImage>&& aCustomCursor,
       const float& aResolutionX, const float& aResolutionY,
       const uint32_t& aHotspotX, const uint32_t& aHotspotY, const bool& aForce);
+  mozilla::ipc::IPCResult RecvUpdateBackgroundColor(const nscolor& aColor);
 
   mozilla::ipc::IPCResult RecvSetLinkStatus(const nsString& aStatus);
 
@@ -967,6 +968,10 @@ class BrowserParent final : public PBrowserParent,
   // tab, it should take this appearance.
   nsIWidget::Cursor mCursor;
 
+#ifdef XP_IOS
+  Maybe<nscolor> mBackgroundColor;
+#endif
+
   nsTArray<nsString> mVerifyDropLinks;
 
 #ifdef DEBUG
