diff --git a/dom/ipc/BrowserChild.cpp b/dom/ipc/BrowserChild.cpp
index 36ec79dc9e3f..5ded5c24518d 100644
--- a/dom/ipc/BrowserChild.cpp
+++ b/dom/ipc/BrowserChild.cpp
@@ -229,6 +229,13 @@ PresShell* BrowserChild::GetTopLevelPresShell() const {
   return nullptr;
 }
 
+void BrowserChild::UpdateBackgroundColor(const nscolor& aColor) {
+  if (!IPCOpen() || mDestroyed) {
+    return;
+  }
+  SendUpdateBackgroundColor(aColor);
+}
+
 bool BrowserChild::UpdateFrame(const RepaintRequest& aRequest) {
   MOZ_ASSERT(aRequest.GetScrollId() != ScrollableLayerGuid::NULL_SCROLL_ID);
 
