diff --git a/dom/ipc/ContentParent.h b/dom/ipc/ContentParent.h
index 1f7b097d0022..8756c25e4543 100644
--- a/dom/ipc/ContentParent.h
+++ b/dom/ipc/ContentParent.h
@@ -732,6 +732,9 @@ class ContentParent final : public PContentParent,
 
   bool WasA11yEverActivated() const { return mWasA11yEverActivated; }
 
+  mozilla::ipc::IPCResult RecvNotifyGPUProcessLost(
+      const uint64_t& aProcessToken);
+
  protected:
   bool CheckBrowsingContextEmbedder(CanonicalBrowsingContext* aBC,
                                     const char* aOperation) const;
