diff --git a/dom/ipc/ContentChild.h b/dom/ipc/ContentChild.h
index 8c9896a71f94..b7418b4ad0bd 100644
--- a/dom/ipc/ContentChild.h
+++ b/dom/ipc/ContentChild.h
@@ -191,14 +191,14 @@ class ContentChild final : public PContentChild,
       Endpoint<PImageBridgeChild>&& aImageBridge,
       Endpoint<PVRManagerChild>&& aVRBridge,
       Endpoint<PRemoteMediaManagerChild>&& aVideoManager,
-      nsTArray<uint32_t>&& namespaces);
+      nsTArray<uint32_t>&& namespaces, const uint64_t& aProcessToken);
 
   mozilla::ipc::IPCResult RecvReinitRendering(
       Endpoint<PCompositorManagerChild>&& aCompositor,
       Endpoint<PImageBridgeChild>&& aImageBridge,
       Endpoint<PVRManagerChild>&& aVRBridge,
       Endpoint<PRemoteMediaManagerChild>&& aVideoManager,
-      nsTArray<uint32_t>&& namespaces);
+      nsTArray<uint32_t>&& namespaces, const uint64_t& aProcessToken);
 
   mozilla::ipc::IPCResult RecvReinitRenderingForDeviceReset();
 
