diff --git a/dom/ipc/ContentParent.h b/dom/ipc/ContentParent.h
index 7fc464ceb107..cbc3bd78bc4c 100644
--- a/dom/ipc/ContentParent.h
+++ b/dom/ipc/ContentParent.h
@@ -744,6 +744,9 @@ class ContentParent final : public PContentParent,
       const OriginAttributes& aOriginAttributes, uint64_t aInnerWindowId,
       const nsCString& aPartitionKey, BlobURLDataRequestResolver&& aResolver);
 
+  mozilla::ipc::IPCResult RecvNotifyGPUProcessLost(
+      const uint64_t& aProcessToken);
+
  protected:
   bool CheckBrowsingContextEmbedder(CanonicalBrowsingContext* aBC,
                                     const char* aOperation) const;
