diff --git a/dom/ipc/ContentChild.h b/dom/ipc/ContentChild.h
index d7465a894df4..72efa8ea5c5d 100644
--- a/dom/ipc/ContentChild.h
+++ b/dom/ipc/ContentChild.h
@@ -167,14 +167,14 @@ class ContentChild final : public PContentChild,
       Endpoint<PImageBridgeChild>&& aImageBridge,
       Endpoint<PVRManagerChild>&& aVRBridge,
       Endpoint<PRemoteMediaManagerChild>&& aVideoManager,
-      nsTArray<uint32_t>&& namespaces);
+      nsTArray<uint32_t>&& namespaces, const uint64_t& aProcessToken);
 
   mozilla::ipc::IPCResult RecvReinitRendering(
       Endpoint<PCompositorManagerChild>&& aCompositor,
       Endpoint<PImageBridgeChild>&& aImageBridge,
       Endpoint<PVRManagerChild>&& aVRBridge,
       Endpoint<PRemoteMediaManagerChild>&& aVideoManager,
-      nsTArray<uint32_t>&& namespaces);
+      nsTArray<uint32_t>&& namespaces, const uint64_t& aProcessToken);
 
   mozilla::ipc::IPCResult RecvReinitRenderingForDeviceReset();
 
