diff --git a/dom/ipc/ContentChild.h b/dom/ipc/ContentChild.h
index 65d223cfe411..0312d840146d 100644
--- a/dom/ipc/ContentChild.h
+++ b/dom/ipc/ContentChild.h
@@ -192,14 +192,14 @@ class ContentChild final : public PContentChild,
       Endpoint<PImageBridgeChild>&& aImageBridge,
       Endpoint<PVRManagerChild>&& aVRBridge,
       Endpoint<PRemoteMediaManagerChild>&& aVideoManager,
-      nsTArray<uint32_t>&& namespaces);
+      nsTArray<uint32_t>&& namespaces, const uint64_t& aProcessToken);
 
   mozilla::ipc::IPCResult RecvReinitRendering(
       Endpoint<PCompositorManagerChild>&& aCompositor,
       Endpoint<PImageBridgeChild>&& aImageBridge,
       Endpoint<PVRManagerChild>&& aVRBridge,
       Endpoint<PRemoteMediaManagerChild>&& aVideoManager,
-      nsTArray<uint32_t>&& namespaces);
+      nsTArray<uint32_t>&& namespaces, const uint64_t& aProcessToken);
 
   mozilla::ipc::IPCResult RecvReinitRenderingForDeviceReset();
 
