diff --git a/js/src/jit/CodeGenerator.cpp b/js/src/jit/CodeGenerator.cpp
index 48bd4aecbca0..e062a6b1a987 100644
--- a/js/src/jit/CodeGenerator.cpp
+++ b/js/src/jit/CodeGenerator.cpp
@@ -4923,7 +4923,7 @@ void CodeGenerator::visitMegamorphicLoadSlotPermissive(
   masm.movePropertyKey(lir->mir()->name(), temp1);
   pushArg(temp2);
   pushArg(temp1);
-  pushArg(obj);
+  pushArg(temp3);
 
   using Fn = bool (*)(JSContext*, HandleObject, HandleId,
                       MegamorphicCacheEntry*, MutableHandleValue);
@@ -5064,7 +5064,11 @@ void CodeGenerator::visitMegamorphicLoadSlotByValuePermissive(
 
   pushArg(temp2);
   pushArg(idVal);
+#ifndef JS_CODEGEN_X86
+  pushArg(temp3);
+#else
   pushArg(obj);
+#endif
 
   using Fn = bool (*)(JSContext*, HandleObject, HandleValue,
                       MegamorphicCacheEntry*, MutableHandleValue);
