diff --git a/js/src/jit/CodeGenerator.cpp b/js/src/jit/CodeGenerator.cpp
index 20781e653162..d0d89959a3a2 100644
--- a/js/src/jit/CodeGenerator.cpp
+++ b/js/src/jit/CodeGenerator.cpp
@@ -4970,7 +4970,7 @@ void CodeGenerator::visitMegamorphicLoadSlotPermissive(
   masm.movePropertyKey(lir->mir()->name(), temp1);
   pushArg(temp2);
   pushArg(temp1);
-  pushArg(obj);
+  pushArg(temp3);
 
   using Fn = bool (*)(JSContext*, HandleObject, HandleId,
                       MegamorphicCacheEntry*, MutableHandleValue);
@@ -5111,7 +5111,11 @@ void CodeGenerator::visitMegamorphicLoadSlotByValuePermissive(
 
   pushArg(temp2);
   pushArg(idVal);
+#ifndef JS_CODEGEN_X86
+  pushArg(temp3);
+#else
   pushArg(obj);
+#endif
 
   using Fn = bool (*)(JSContext*, HandleObject, HandleValue,
                       MegamorphicCacheEntry*, MutableHandleValue);
