diff --git a/js/src/jit/IonCacheIRCompiler.h b/js/src/jit/IonCacheIRCompiler.h
index 8d5e0d5a3c3e..f390a8b59b74 100644
--- a/js/src/jit/IonCacheIRCompiler.h
+++ b/js/src/jit/IonCacheIRCompiler.h
@@ -57,6 +57,7 @@ class MOZ_RAII IonCacheIRCompiler : public CacheIRCompiler {
   const CacheIRWriter& writer_;
   IonIC* ic_;
   IonScript* ionScript_;
+  IonICStub* currentStub_;
 
   Vector<CodeOffset, 4, SystemAllocPolicy> nextCodeOffsets_;
   mozilla::Maybe<LiveRegisterSet> liveRegs_;
@@ -96,6 +97,7 @@ class MOZ_RAII IonCacheIRCompiler : public CacheIRCompiler {
                                 StringCharOutOfBounds outOfBounds);
 
   void pushStubCodePointer();
+  void pushStubAndStubCodePointer();
 
   CACHE_IR_COMPILER_UNSHARED_GENERATED
 };
