diff --git a/docshell/base/CanonicalBrowsingContext.cpp b/docshell/base/CanonicalBrowsingContext.cpp
index 1d0b9fec6ed9..c37d74cd0124 100644
--- a/docshell/base/CanonicalBrowsingContext.cpp
+++ b/docshell/base/CanonicalBrowsingContext.cpp
@@ -319,6 +319,10 @@ void CanonicalBrowsingContext::ReplacedBy(
   txn.SetAllowJavascript(GetAllowJavascript());
   txn.SetForceEnableTrackingProtection(GetForceEnableTrackingProtection());
   txn.SetUserAgentOverride(GetUserAgentOverride());
+  txn.SetPlatformOverride(GetPlatformOverride());
+  txn.SetAppVersionOverride(GetAppVersionOverride());
+  txn.SetOscpuOverride(GetOscpuOverride());
+  txn.SetBuildIDOverride(GetBuildIDOverride());
   txn.SetSuspendMediaWhenInactive(GetSuspendMediaWhenInactive());
   txn.SetDisplayMode(GetDisplayMode());
   txn.SetForceDesktopViewport(GetForceDesktopViewport());
