diff --git a/layout/painting/nsCSSRendering.cpp b/layout/painting/nsCSSRendering.cpp
index 70f2881b021e..45134a48ffbc 100644
--- a/layout/painting/nsCSSRendering.cpp
+++ b/layout/painting/nsCSSRendering.cpp
@@ -2844,11 +2844,11 @@ nsRect nsCSSRendering::ComputeImageLayerPositioningArea(
       }
 
       // If we have the dynamic toolbar, we need to expand the image area to
-      // include the region under the dynamic toolbar, otherwise we will see a
-      // blank space under the toolbar.
+      // the large viewport, which includes the region the toolbar can reveal,
+      // otherwise we will see a blank space under the toolbar.
       if (aPresContext->IsRootContentDocumentCrossProcess() &&
           aPresContext->HasDynamicToolbar()) {
-        positionArea.SizeTo(nsLayoutUtils::ExpandHeightForDynamicToolbar(
+        positionArea.SizeTo(nsLayoutUtils::ExpandHeightForViewportUnits(
             aPresContext, positionArea.Size()));
       }
     }
