diff --git a/layout/painting/nsDisplayList.h b/layout/painting/nsDisplayList.h
index 7c11e40845cf..b4df1dde44cf 100644
--- a/layout/painting/nsDisplayList.h
+++ b/layout/painting/nsDisplayList.h
@@ -5922,6 +5922,7 @@ class nsDisplayAsyncZoom final : public nsDisplayOwnLayer {
 
   NS_DISPLAY_DECL_NAME("AsyncZoom", TYPE_ASYNC_ZOOM)
 
+  nsRect GetBounds(nsDisplayListBuilder* aBuilder, bool* aSnap) const override;
   void HitTest(nsDisplayListBuilder* aBuilder, const nsRect& aRect,
                HitTestState* aState, nsTArray<nsIFrame*>* aOutFrames) override;
   bool UpdateScrollData(layers::WebRenderScrollData* aData,
