diff --git a/layout/painting/nsDisplayList.h b/layout/painting/nsDisplayList.h
index fcf53b8617f4..15ebf527c648 100644
--- a/layout/painting/nsDisplayList.h
+++ b/layout/painting/nsDisplayList.h
@@ -5921,6 +5921,7 @@ class nsDisplayAsyncZoom final : public nsDisplayOwnLayer {
 
   NS_DISPLAY_DECL_NAME("AsyncZoom", TYPE_ASYNC_ZOOM)
 
+  nsRect GetBounds(nsDisplayListBuilder* aBuilder, bool* aSnap) const override;
   void HitTest(nsDisplayListBuilder* aBuilder, const nsRect& aRect,
                HitTestState* aState, nsTArray<nsIFrame*>* aOutFrames) override;
   bool UpdateScrollData(layers::WebRenderScrollData* aData,
