diff --git a/hal/moz.build b/hal/moz.build
index bb486ec1e411..d5d951aa54b0 100644
--- a/hal/moz.build
+++ b/hal/moz.build
@@ -85,6 +85,15 @@ elif CONFIG["MOZ_WIDGET_TOOLKIT"] == "cocoa":
         "fallback/FallbackScreenConfiguration.cpp",
         "fallback/FallbackVibration.cpp",
     ]
+elif CONFIG["MOZ_WIDGET_TOOLKIT"] == "uikit":
+    UNIFIED_SOURCES += [
+        "fallback/FallbackScreenConfiguration.cpp",
+        "fallback/FallbackVibration.cpp",
+        "uikit/UIKitBattery.mm",
+        "uikit/UIKitHeterogeneousCpuInfo.cpp",
+        "uikit/UIKitProcessPriority.mm",
+        "uikit/UIKitSensor.mm",
+    ]
 elif CONFIG["OS_TARGET"] in ("OpenBSD", "NetBSD", "FreeBSD", "DragonFly"):
     UNIFIED_SOURCES += [
         "fallback/FallbackHeterogeneousCpuInfo.cpp",
