diff --git a/gfx/layers/ipc/CompositorManagerChild.cpp b/gfx/layers/ipc/CompositorManagerChild.cpp
index d6216ce8f20a..dc60b9b3bad7 100644
--- a/gfx/layers/ipc/CompositorManagerChild.cpp
+++ b/gfx/layers/ipc/CompositorManagerChild.cpp
@@ -142,7 +142,7 @@ bool CompositorManagerChild::CreateContentCompositorBridge(
     return false;
   }
 
-  bridge->InitForContent(aNamespace);
+  bridge->InitForContent(aNamespace, sInstance->mProcessToken);
   return true;
 }
 
