diff --git a/gfx/layers/ipc/CompositorBridgeChild.h b/gfx/layers/ipc/CompositorBridgeChild.h
index c2051f3c355e..38b5d448b9ca 100644
--- a/gfx/layers/ipc/CompositorBridgeChild.h
+++ b/gfx/layers/ipc/CompositorBridgeChild.h
@@ -62,7 +62,7 @@ class CompositorBridgeChild final : public PCompositorBridgeChild,
   /**
    * Initialize the singleton compositor bridge for a content process.
    */
-  void InitForContent(uint32_t aNamespace);
+  void InitForContent(uint32_t aNamespace, uint64_t aProcessToken);
 
   void InitForWidget(uint64_t aProcessToken, uint32_t aNamespace);
 
