diff --git a/gfx/layers/ipc/UiCompositorControllerParent.cpp b/gfx/layers/ipc/UiCompositorControllerParent.cpp
index b04c706d6845..fa97dad26de3 100644
--- a/gfx/layers/ipc/UiCompositorControllerParent.cpp
+++ b/gfx/layers/ipc/UiCompositorControllerParent.cpp
@@ -114,14 +114,14 @@ mozilla::ipc::IPCResult UiCompositorControllerParent::RecvMaxToolbarHeight(
 
 mozilla::ipc::IPCResult UiCompositorControllerParent::RecvFixedBottomOffset(
     const int32_t& aOffset) {
-#if defined(MOZ_WIDGET_ANDROID)
+#if defined(MOZ_WIDGET_ANDROID) || defined(MOZ_WIDGET_UIKIT)
   CompositorBridgeParent* parent =
       CompositorBridgeParent::GetCompositorBridgeParentFromLayersId(
           mRootLayerTreeId);
   if (parent) {
     parent->SetFixedLayerMargins(0, aOffset);
   }
-#endif  // defined(MOZ_WIDGET_ANDROID)
+#endif
 
   return IPC_OK();
 }
