diff --git a/gfx/layers/ipc/UiCompositorControllerChild.cpp b/gfx/layers/ipc/UiCompositorControllerChild.cpp
index 89dc6497e3cb..236aab5f17ed 100644
--- a/gfx/layers/ipc/UiCompositorControllerChild.cpp
+++ b/gfx/layers/ipc/UiCompositorControllerChild.cpp
@@ -22,8 +22,10 @@
 #  include "mozilla/widget/AndroidUiThread.h"
 
 static RefPtr<nsThread> GetUiThread() { return mozilla::GetAndroidUiThread(); }
+#elif defined(MOZ_WIDGET_UIKIT)
+static nsCOMPtr<nsIThread> GetUiThread() { return do_GetMainThread(); }
 #else
-static RefPtr<nsThread> GetUiThread() {
+static nsCOMPtr<nsIThread> GetUiThread() {
   MOZ_CRASH("Platform does not support UiCompositorController");
   return nullptr;
 }
@@ -196,29 +198,38 @@ bool UiCompositorControllerChild::EnableLayerUpdateNotifications(
 void UiCompositorControllerChild::Destroy() {
   MOZ_ASSERT(NS_IsMainThread());
 
+  auto destroy = [this]() {
+    MOZ_ASSERT(GetUiThread()->IsOnCurrentThread());
+
+    // Clear the process token so that we don't notify the GPUProcessManager
+    // about an abnormal shutdown, thereby tearing down the GPU process.
+    mProcessToken = 0;
+
+    if (mWidget) {
+      // Dispatch mWidget to main thread to prevent it from being destructed
+      // by the ui thread.
+      RefPtr<nsIWidget> widget = std::move(mWidget);
+      NS_ReleaseOnMainThread("UiCompositorControllerChild::mWidget",
+                             widget.forget());
+    }
+
+    if (mIsOpen) {
+      // Close the underlying IPC channel.
+      PUiCompositorControllerChild::Close();
+      mIsOpen = false;
+    }
+  };
+
+  if (GetUiThread()->IsOnCurrentThread()) {
+    destroy();
+    return;
+  }
+
   layers::SynchronousTask task("UiCompositorControllerChild::Destroy");
   GetUiThread()->Dispatch(NS_NewRunnableFunction(
-      "layers::UiCompositorControllerChild::Destroy", [&]() {
-        MOZ_ASSERT(GetUiThread()->IsOnCurrentThread());
+      "layers::UiCompositorControllerChild::Destroy", [destroy, &task]() {
         AutoCompleteTask complete(&task);
-
-        // Clear the process token so that we don't notify the GPUProcessManager
-        // about an abnormal shutdown, thereby tearing down the GPU process.
-        mProcessToken = 0;
-
-        if (mWidget) {
-          // Dispatch mWidget to main thread to prevent it from being destructed
-          // by the ui thread.
-          RefPtr<nsIWidget> widget = std::move(mWidget);
-          NS_ReleaseOnMainThread("UiCompositorControllerChild::mWidget",
-                                 widget.forget());
-        }
-
-        if (mIsOpen) {
-          // Close the underlying IPC channel.
-          PUiCompositorControllerChild::Close();
-          mIsOpen = false;
-        }
+        destroy();
       }));
 
   task.Wait();
