diff --git a/gfx/layers/ipc/WebRenderMessages.ipdlh b/gfx/layers/ipc/WebRenderMessages.ipdlh
index 212594007ab3..96df56cb144f 100644
--- a/gfx/layers/ipc/WebRenderMessages.ipdlh
+++ b/gfx/layers/ipc/WebRenderMessages.ipdlh
@@ -162,6 +162,8 @@ struct OpAddRawFont {
 struct OpAddFontDescriptor {
   OffsetRange bytes;
   uint32_t fontIndex;
+  bool isSystemFont;
+  uint32_t systemFontTraits;
   FontKey key;
 };
 
