diff --git a/gfx/ipc/GPUProcessManager.cpp b/gfx/ipc/GPUProcessManager.cpp
index 74b0ec077b67..164d5317199f 100644
--- a/gfx/ipc/GPUProcessManager.cpp
+++ b/gfx/ipc/GPUProcessManager.cpp
@@ -1378,7 +1378,9 @@ bool GPUProcessManager::CreateContentBridges(
     ipc::Endpoint<PImageBridgeChild>* aOutImageBridge,
     ipc::Endpoint<PVRManagerChild>* aOutVRBridge,
     ipc::Endpoint<PRemoteMediaManagerChild>* aOutVideoManager,
-    dom::ContentParentId aChildId, nsTArray<uint32_t>* aNamespaces) {
+    dom::ContentParentId aChildId, nsTArray<uint32_t>* aNamespaces,
+    uint64_t* aOutProcessToken) {
+  MOZ_ASSERT(aOutProcessToken);
   const uint32_t cmNamespace = AllocateNamespace();
   if (!CreateContentCompositorManager(aOtherProcess, aChildId, cmNamespace,
                                       aOutCompositor) ||
@@ -1386,6 +1388,9 @@ bool GPUProcessManager::CreateContentBridges(
       !CreateContentVRManager(aOtherProcess, aChildId, aOutVRBridge)) {
     return false;
   }
+  // REYNARD: Publish the current GPU process token so content can report
+  // compositor abnormal shutdowns for the correct process instance.
+  *aOutProcessToken = mProcessToken;
   // RemoteMediaManager is only supported in the GPU process, so we allow this
   // to be fallible.
   CreateContentRemoteMediaManager(aOtherProcess, aChildId, aOutVideoManager);
