diff --git a/gfx/ipc/GPUProcessManager.h b/gfx/ipc/GPUProcessManager.h
index b2e3e5e01127..7b6b22af3b6c 100644
--- a/gfx/ipc/GPUProcessManager.h
+++ b/gfx/ipc/GPUProcessManager.h
@@ -123,7 +123,8 @@ class GPUProcessManager final : public GPUProcessHost::Listener {
       mozilla::ipc::Endpoint<PImageBridgeChild>* aOutImageBridge,
       mozilla::ipc::Endpoint<PVRManagerChild>* aOutVRBridge,
       mozilla::ipc::Endpoint<PRemoteMediaManagerChild>* aOutVideoManager,
-      dom::ContentParentId aChildId, nsTArray<uint32_t>* aNamespaces);
+      dom::ContentParentId aChildId, nsTArray<uint32_t>* aNamespaces,
+      uint64_t* aOutProcessToken);
 
   nsresult CreateRddVideoBridge(RDDProcessManager* aRDD, RDDChild* aChild);
 #ifdef MOZ_WMF_MEDIA_ENGINE
