diff --git a/gfx/ipc/GPUProcessManager.cpp b/gfx/ipc/GPUProcessManager.cpp
index 8429c88bf057..63b1a78fa8a6 100644
--- a/gfx/ipc/GPUProcessManager.cpp
+++ b/gfx/ipc/GPUProcessManager.cpp
@@ -1384,7 +1384,9 @@ bool GPUProcessManager::CreateContentBridges(
     ipc::Endpoint<PImageBridgeChild>* aOutImageBridge,
     ipc::Endpoint<PVRManagerChild>* aOutVRBridge,
     ipc::Endpoint<PRemoteMediaManagerChild>* aOutVideoManager,
-    dom::ContentParentId aChildId, nsTArray<uint32_t>* aNamespaces) {
+    dom::ContentParentId aChildId, nsTArray<uint32_t>* aNamespaces,
+    uint64_t* aOutProcessToken) {
+  MOZ_ASSERT(aOutProcessToken);
   const uint32_t compositorManagerNamespace = AllocateNamespace();
   const uint32_t compositorBridgeNamespace = AllocateNamespace();
   const uint32_t imageBridgeNamespace = AllocateNamespace();
@@ -1398,6 +1400,9 @@ bool GPUProcessManager::CreateContentBridges(
                               aOutVRBridge)) {
     return false;
   }
+  // REYNARD: Publish the current GPU process token so content can report
+  // compositor abnormal shutdowns for the correct process instance.
+  *aOutProcessToken = mProcessToken;
   // RemoteMediaManager is only supported in the GPU process, so we allow this
   // to be fallible.
   CreateContentRemoteMediaManager(aOtherProcess, aChildId, aOutVideoManager);
