diff --git a/xpcom/base/moz.build b/xpcom/base/moz.build
index 4ae093061f5f..be5030f4868c 100644
--- a/xpcom/base/moz.build
+++ b/xpcom/base/moz.build
@@ -225,11 +225,22 @@ if CONFIG["OS_TARGET"] == "WINNT":
     ]
 
 if CONFIG["OS_TARGET"] == "Darwin":
+    EXPORTS.mozilla += [
+        "MemoryPressureLevelMac.h",
+    ]
+    if CONFIG["MOZ_WIDGET_TOOLKIT"] != "uikit":
+        UNIFIED_SOURCES += [
+            "AvailableMemoryWatcherMac.cpp",
+        ]
+
+if CONFIG["MOZ_WIDGET_TOOLKIT"] == "uikit":
     UNIFIED_SOURCES += [
-        "AvailableMemoryWatcherMac.cpp",
+        "AvailableMemoryWatcherIOS.cpp",
+        "MemoryPressureMonitorIOS.cpp",
     ]
     EXPORTS.mozilla += [
-        "MemoryPressureLevelMac.h",
+        "MemoryPressureLevelIOS.h",
+        "MemoryPressureMonitorIOS.h",
     ]
 
 if CONFIG["OS_TARGET"] == "Linux":
