diff --git a/xpcom/base/moz.build b/xpcom/base/moz.build
index 861a4fbff37b..12d2d20e5858 100644
--- a/xpcom/base/moz.build
+++ b/xpcom/base/moz.build
@@ -222,11 +222,20 @@ if CONFIG["OS_TARGET"] == "WINNT":
     ]
 
 if CONFIG["OS_TARGET"] == "Darwin":
+    EXPORTS.mozilla += [
+        "MemoryPressureLevelMac.h",
+    ]
+    if CONFIG["MOZ_WIDGET_TOOLKIT"] != "uikit":
+        UNIFIED_SOURCES += [
+            "AvailableMemoryWatcherMac.cpp",
+        ]
+
+if CONFIG["MOZ_WIDGET_TOOLKIT"] == "uikit":
     UNIFIED_SOURCES += [
-        "AvailableMemoryWatcherMac.cpp",
+        "AvailableMemoryWatcherIOS.cpp",
     ]
     EXPORTS.mozilla += [
-        "MemoryPressureLevelMac.h",
+        "MemoryPressureLevelIOS.h",
     ]
 
 if CONFIG["OS_TARGET"] == "Linux":
