diff --git a/dom/base/Navigator.h b/dom/base/Navigator.h
index d303ab778fb5..1b654d0de50e 100644
--- a/dom/base/Navigator.h
+++ b/dom/base/Navigator.h
@@ -160,6 +160,12 @@ class Navigator final : public nsISupports, public nsWrapperCache {
   // Navigator_Binding::ClearCachedPlatformValue(this);
   void ClearPlatformCache();
 
+  void ClearAppVersionCache();
+
+  void ClearOscpuCache();
+
+  void ClearBuildIDCache();
+
   // Clears the user agent cache by calling:
   // Navigator_Binding::ClearCachedUserAgentValue(this);
   void ClearUserAgentCache();
