diff --git a/devtools/shared/system.js b/devtools/shared/system.js
index 65ee9cca2b3d..ad4bcd1ba832 100644
--- a/devtools/shared/system.js
+++ b/devtools/shared/system.js
@@ -61,13 +61,18 @@ function getSystemInfo() {
   const os = appInfo.OS;
   version = appInfo.version;
 
-  const bundle = Services.strings.createBundle(
-    "chrome://branding/locale/brand.properties"
-  );
-  if (bundle) {
-    brandName = bundle.GetStringFromName("brandFullName");
+  // FIXME: Missing branding bundle?
+  if (lazy.AppConstants.platform === "ios") {
+    brandName = appInfo.name;
   } else {
-    brandName = null;
+    const bundle = Services.strings.createBundle(
+      "chrome://branding/locale/brand.properties"
+    );
+    if (bundle) {
+      brandName = bundle.GetStringFromName("brandFullName");
+    } else {
+      brandName = null;
+    }
   }
 
   if (win) {
