diff --git a/widget/uikit/components.conf b/widget/uikit/components.conf
index e8f65c05c3e0..d804e0aa2253 100644
--- a/widget/uikit/components.conf
+++ b/widget/uikit/components.conf
@@ -31,6 +31,15 @@ Classes = [
         'type': 'nsIOSBridge',
         'headers': ['/widget/uikit/IOSBridge.h'],
     },
+    {
+        'cid': '{f0cd2511-0911-4684-bdc6-117bc98f66c8}',
+        'contract_ids': ['@mozilla.org/uriloader/external-helper-app-service;1'],
+        'interfaces': ['nsIExternalHelperAppService'],
+        'type': 'ExternalResponseService',
+        'headers': ['/widget/uikit/ExternalResponseService.h'],
+        'constructor': 'ExternalResponseService::GetSingleton',
+        'processes': ProcessSelector.MAIN_PROCESS_ONLY,
+    },
     {
         'cid': '{2d96b3df-c051-11d1-a827-0040959a28c9}',
         'contract_ids': ['@mozilla.org/widget/appshell/uikit;1'],
