diff --git a/widget/uikit/moz.build b/widget/uikit/moz.build
index f5d8821773c0..5ead3454cf7e 100644
--- a/widget/uikit/moz.build
+++ b/widget/uikit/moz.build
@@ -8,35 +8,50 @@ with Files("**"):
 with Files("*TextInput*"):
     BUG_COMPONENT = ("Core", "DOM: UI Events & Focus Handling")
 
+with Files("GeckoEditableSupport.*"):
+    BUG_COMPONENT = ("Core", "DOM: UI Events & Focus Handling")
+
 EXPORTS += [
     "IOSBridge.h",
 ]
 
 EXPORTS.mozilla.widget += [
+    "CompositorWidgetChild.h",
+    "CompositorWidgetParent.h",
     "EventDispatcher.h",
+    "GeckoViewRuntimeSupport.h",
     "GeckoViewSupport.h",
     "IOSView.h",
     "nsWindow.h",
+    "UIKitCompositorWidget.h",
 ]
 
 EXPORTS.GeckoView += [
+    "GeckoViewRuntimeSupport.h",
     "GeckoViewSwiftSupport.h",
 ]
 
 SOURCES += [
+    "CompositorWidgetChild.cpp",
+    "CompositorWidgetParent.cpp",
     "EventDispatcher.mm",
+    "ExternalResponseService.mm",
+    "GeckoEditableSupport.mm",
     "GfxInfo.cpp",
     "IOSBridge.mm",
     "MediaKeysEventSourceFactory.cpp",
     "nsAppShell.mm",
     "nsBidiKeyboard.mm",
+    "nsClipboard.mm",
     "nsLookAndFeel.mm",
     "nsNativeThemeUIKit.mm",
     "nsUserIdleServiceUIKit.cpp",
     "nsWidgetFactory.mm",
     "nsWindow.mm",
+    "PictureInPictureSupport.mm",
     "ScreenHelperUIKit.mm",
     "TextInputHandler.mm",
+    "UIKitCompositorWidget.cpp",
     "UIKitUtils.mm",
 ]
 
