diff --git a/widget/uikit/nsWindow.h b/widget/uikit/nsWindow.h
index dd2d9e2be070..87b679d1c6c4 100644
--- a/widget/uikit/nsWindow.h
+++ b/widget/uikit/nsWindow.h
@@ -21,11 +21,14 @@ typedef struct objc_object ChildView;
 #endif
 
 namespace mozilla::layers {
+struct APZHandledResult;
 class NativeLayerRootCA;
-}
+class NativeLayerRootRemoteMacParent;
+}  // namespace mozilla::layers
 
 namespace mozilla::widget {
 class EventDispatcher;
+class PlatformCompositorWidgetDelegate;
 class TextInputHandler;
 }  // namespace mozilla::widget
 
@@ -55,12 +58,16 @@ class nsWindow final : public nsIWidget {
   LayoutDeviceIntPoint WidgetToScreenOffset() override;
 
   void SetBackgroundColor(const nscolor& aColor) override;
+  void SetCursor(const Cursor& aCursor) override;
   void* GetNativeData(uint32_t aDataType) override;
 
   void Move(const DesktopPoint&) override;
   nsSizeMode SizeMode() override { return mSizeMode; }
   void SetSizeMode(nsSizeMode aMode) override;
-  void EnteredFullScreen(bool aFullScreen);
+  [[nodiscard]] nsresult MakeFullScreen(bool aFullScreen) override
+      MOZ_REQUIRES(mozilla::sMainThreadCapability);
+  void EnteredFullScreen(bool aFullScreen)
+      MOZ_REQUIRES(mozilla::sMainThreadCapability);
   void Resize(const DesktopSize&, bool aRepaint) override;
   void Resize(const DesktopRect&, bool aRepaint) override;
   void DoResize(double aX, double aY, double aWidth, double aHeight,
@@ -77,6 +84,7 @@ class nsWindow final : public nsIWidget {
     // XXX: terrible
     return 326.0f;
   }
+  uint32_t GetMaxTouchPoints() const override;
   double GetDefaultScaleInternal() override { return BackingScaleFactor(); }
   int32_t RoundsWidgetCoordinatesTo() override;
 
@@ -93,6 +101,9 @@ class nsWindow final : public nsIWidget {
   void SetInputContext(const InputContext& aContext,
                        const InputContextAction& aAction) override;
   InputContext GetInputContext() override;
+  void NotifyWindowActivationState(bool aActive);
+  void SetCompositorWidgetDelegate(
+      mozilla::widget::CompositorWidgetDelegate* aDelegate) override;
   TextEventDispatcherListener* GetNativeTextEventDispatcherListener() override;
 
   mozilla::widget::TextInputHandler* GetTextInputHandler() const {
@@ -109,6 +120,11 @@ class nsWindow final : public nsIWidget {
   */
 
   mozilla::layers::NativeLayerRoot* GetNativeLayerRoot() override;
+  void GetCompositorWidgetInitData(
+      mozilla::widget::CompositorWidgetInitData* aInitData) override;
+  void DestroyCompositor() override;
+  void NotifyCompositorSessionLost(
+      mozilla::layers::CompositorSession* aSession) override;
 
   void HandleMainThreadCATransaction();
 
@@ -138,6 +154,16 @@ class nsWindow final : public nsIWidget {
     mIOSView = aView;
   }
   mozilla::widget::IOSView* GetIOSView() const { return mIOSView; }
+  nsCursor GetCursorType() const { return mCursor.mDefaultCursor; }
+
+  bool DispatchWindowEvent(mozilla::WidgetGUIEvent& aEvent) override;
+  void DispatchPanGestureInput(mozilla::PanGestureInput& aInput);
+  using nsIWidget::DispatchPinchGestureInput;
+  void DispatchTouchStart(mozilla::WidgetTouchEvent* aEvent,
+                          uint64_t aSequenceId);
+  void NotifyAPZInputResult(
+      uint64_t aSequenceId,
+      const mozilla::layers::APZHandledResult& aHandledResult);
 
  protected:
   virtual ~nsWindow();
@@ -147,6 +173,8 @@ class nsWindow final : public nsIWidget {
   nsresult GetCurrentOffset(uint32_t& aOffset, uint32_t& aLength);
   nsresult DeleteRange(int aOffset, int aLen);
 
+  void CloseNativeLayerRemoteParent(bool aWaitUntilClosed);
+  void DestroyNativeLayerRoot();
   void TearDownView();
 
   ChildView* mNativeView;
@@ -158,8 +186,12 @@ class nsWindow final : public nsIWidget {
   mozilla::widget::InputContext mInputContext;
   RefPtr<mozilla::widget::TextInputHandler> mTextInputHandler;
   RefPtr<mozilla::widget::IOSView> mIOSView;
+  mozilla::widget::PlatformCompositorWidgetDelegate* mCompositorWidgetDelegate =
+      nullptr;
 
   RefPtr<mozilla::layers::NativeLayerRootCA> mNativeLayerRoot;
+  RefPtr<mozilla::layers::NativeLayerRootRemoteMacParent>
+      mNativeLayerRootRemoteMacParent;
 
   RefPtr<mozilla::CancelableRunnable> mUnsuspendAsyncCATransactionsRunnable;
   LayoutDeviceIntRect mBounds;
