diff --git a/widget/uikit/PictureInPictureSupport.h b/widget/uikit/PictureInPictureSupport.h
new file mode 100644
index 000000000000..66eb810b9454
--- /dev/null
+++ b/widget/uikit/PictureInPictureSupport.h
@@ -0,0 +1,23 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/* This source code is introduced during the work of an iOS 13+ browser
+ * https://github.com/minh-ton/reynard-browser in Feb 2026. */
+
+#ifndef mozilla_widget_PictureInPictureSupport_h
+#define mozilla_widget_PictureInPictureSupport_h
+
+#include <GeckoView/GeckoViewSwiftSupport.h>
+
+class nsWindow;
+
+namespace mozilla::widget {
+
+void InstallPictureInPictureLayerObserver(nsWindow* aWindow);
+GeckoViewPictureInPictureCandidateList* GetPictureInPictureCandidates(
+    nsWindow* aWindow);
+
+}  // namespace mozilla::widget
+
+#endif  // mozilla_widget_PictureInPictureSupport_h
