diff --git a/third_party/libwebrtc/moz.build b/third_party/libwebrtc/moz.build
index 70bd75ee2378..909917caac36 100644
--- a/third_party/libwebrtc/moz.build
+++ b/third_party/libwebrtc/moz.build
@@ -547,9 +547,6 @@ if CONFIG["OS_TARGET"] == "Darwin":
 
     DIRS += [
         "/third_party/libwebrtc/common_audio/fir_filter_gn",
-        "/third_party/libwebrtc/modules/desktop_capture/desktop_capture_gn",
-        "/third_party/libwebrtc/modules/desktop_capture/desktop_capture_objc_gn",
-        "/third_party/libwebrtc/modules/desktop_capture/primitives_gn",
         "/third_party/libwebrtc/rtc_base/system/cocoa_threading_gn",
         "/third_party/libwebrtc/rtc_base/system/gcd_helpers_gn",
         "/third_party/libwebrtc/sdk/base_objc_gn",
@@ -558,6 +555,13 @@ if CONFIG["OS_TARGET"] == "Darwin":
         "/third_party/libwebrtc/sdk/videoframebuffer_objc_gn"
     ]
 
+    if CONFIG["MOZ_WIDGET_TOOLKIT"] != "uikit":
+        DIRS += [
+            "/third_party/libwebrtc/modules/desktop_capture/desktop_capture_gn",
+            "/third_party/libwebrtc/modules/desktop_capture/desktop_capture_objc_gn",
+            "/third_party/libwebrtc/modules/desktop_capture/primitives_gn",
+        ]
+
 if CONFIG["OS_TARGET"] == "OpenBSD":
 
     DIRS += [
