diff --git a/dom/media/platforms/apple/AppleVTDecoder.cpp b/dom/media/platforms/apple/AppleVTDecoder.cpp
index eb9a080f..f33fdf1b 100644
--- a/dom/media/platforms/apple/AppleVTDecoder.cpp
+++ b/dom/media/platforms/apple/AppleVTDecoder.cpp
@@ -712,22 +712,24 @@ MediaResult AppleVTDecoder::InitializeSession() {
                        RESULT_DETAIL("Couldn't create decompression session!"));
   }
 
-  CFBooleanRef isUsingHW = nullptr;
-  rv = VTSessionCopyProperty(
-      mSession,
-      kVTDecompressionPropertyKey_UsingHardwareAcceleratedVideoDecoder,
-      kCFAllocatorDefault, &isUsingHW);
-  if (rv == noErr) {
-    mIsHardwareAccelerated = isUsingHW == kCFBooleanTrue;
-    LOG("AppleVTDecoder: {} hardware accelerated decoding",
-        mIsHardwareAccelerated ? "using" : "not using");
-  } else {
-    LOG("AppleVTDecoder: maybe hardware accelerated decoding "
-        "(VTSessionCopyProperty query failed {})",
-        static_cast<int>(rv));
-  }
-  if (isUsingHW) {
-    CFRelease(isUsingHW);
+  if (__builtin_available(macos 10.13, iOS 17.0, *)) {
+    CFBooleanRef isUsingHW = nullptr;
+    rv = VTSessionCopyProperty(
+        mSession,
+        kVTDecompressionPropertyKey_UsingHardwareAcceleratedVideoDecoder,
+        kCFAllocatorDefault, &isUsingHW);
+    if (rv == noErr) {
+      mIsHardwareAccelerated = isUsingHW == kCFBooleanTrue;
+      LOG("AppleVTDecoder: {} hardware accelerated decoding",
+          mIsHardwareAccelerated ? "using" : "not using");
+    } else {
+      LOG("AppleVTDecoder: maybe hardware accelerated decoding "
+          "(VTSessionCopyProperty query failed {})",
+          static_cast<int>(rv));
+    }
+    if (isUsingHW) {
+      CFRelease(isUsingHW);
+    }
   }
 
   return NS_OK;
@@ -774,20 +776,26 @@ CFDictionaryRef AppleVTDecoder::CreateDecoderExtensions() {
 }
 
 CFDictionaryRef AppleVTDecoder::CreateDecoderSpecification() {
-  const void* specKeys[] = {
-      kVTVideoDecoderSpecification_EnableHardwareAcceleratedVideoDecoder};
-  const void* specValues[1];
-  if (gfx::gfxVars::CanUseHardwareVideoDecoding()) {
-    specValues[0] = kCFBooleanTrue;
-  } else {
-    // This GPU is blacklisted for hardware decoding.
-    specValues[0] = kCFBooleanFalse;
+  if (__builtin_available(macos 10.13, iOS 17.0, *)) {
+    const void* specKeys[] = {
+        kVTVideoDecoderSpecification_EnableHardwareAcceleratedVideoDecoder};
+    const void* specValues[1];
+    if (gfx::gfxVars::CanUseHardwareVideoDecoding()) {
+      specValues[0] = kCFBooleanTrue;
+    } else {
+      // This GPU is blacklisted for hardware decoding.
+      specValues[0] = kCFBooleanFalse;
+    }
+    static_assert(std::size(specKeys) == std::size(specValues),
+                  "Non matching keys/values array size");
+
+    return CFDictionaryCreate(
+        kCFAllocatorDefault, specKeys, specValues, std::size(specKeys),
+        &kCFTypeDictionaryKeyCallBacks, &kCFTypeDictionaryValueCallBacks);
   }
-  static_assert(std::size(specKeys) == std::size(specValues),
-                "Non matching keys/values array size");
 
-  return CFDictionaryCreate(kCFAllocatorDefault, specKeys, specValues,
-                            std::size(specKeys), &kCFTypeDictionaryKeyCallBacks,
+  return CFDictionaryCreate(kCFAllocatorDefault, nullptr, nullptr, 0,
+                            &kCFTypeDictionaryKeyCallBacks,
                             &kCFTypeDictionaryValueCallBacks);
 }
 
diff --git a/dom/media/platforms/apple/AppleVTEncoder.cpp b/dom/media/platforms/apple/AppleVTEncoder.cpp
index 3359cbad..e7ccdb1d 100644
--- a/dom/media/platforms/apple/AppleVTEncoder.cpp
+++ b/dom/media/platforms/apple/AppleVTEncoder.cpp
@@ -32,7 +32,7 @@ extern LazyLogModule sPEMLog;
 
 static CFDictionaryRef BuildEncoderSpec(const bool aHardwareNotAllowed,
                                         const bool aLowLatencyRateControl) {
-  if (__builtin_available(macos 11.3, *)) {
+  if (__builtin_available(macos 11.3, iOS 17.4, *)) {
     if (aLowLatencyRateControl) {
       // If doing low-latency rate control, the hardware encoder
       // is required.
@@ -48,14 +48,20 @@ static CFDictionaryRef BuildEncoderSpec(const bool aHardwareNotAllowed,
                                 &kCFTypeDictionaryValueCallBacks);
     }
   }
-  const void* keys[] = {
-      kVTVideoEncoderSpecification_EnableHardwareAcceleratedVideoEncoder};
-  const void* values[] = {aHardwareNotAllowed ? kCFBooleanFalse
-                                              : kCFBooleanTrue};
+  if (__builtin_available(macos 10.13, iOS 17.4, *)) {
+    const void* keys[] = {
+        kVTVideoEncoderSpecification_EnableHardwareAcceleratedVideoEncoder};
+    const void* values[] = {aHardwareNotAllowed ? kCFBooleanFalse
+                                                : kCFBooleanTrue};
 
-  static_assert(std::size(keys) == std::size(values),
-                "Non matching keys/values array size");
-  return CFDictionaryCreate(kCFAllocatorDefault, keys, values, std::size(keys),
+    static_assert(std::size(keys) == std::size(values),
+                  "Non matching keys/values array size");
+    return CFDictionaryCreate(
+        kCFAllocatorDefault, keys, values, std::size(keys),
+        &kCFTypeDictionaryKeyCallBacks, &kCFTypeDictionaryValueCallBacks);
+  }
+
+  return CFDictionaryCreate(kCFAllocatorDefault, nullptr, nullptr, 0,
                             &kCFTypeDictionaryKeyCallBacks,
                             &kCFTypeDictionaryValueCallBacks);
 }
@@ -84,7 +90,7 @@ bool AppleVTEncoder::SetAverageBitrate(uint32_t aBitsPerSec) {
 bool AppleVTEncoder::SetConstantBitrate(uint32_t aBitsPerSec) {
   MOZ_ASSERT(mSession);
 
-  if (__builtin_available(macos 13.0, *)) {
+  if (__builtin_available(macos 13.0, iOS 16.0, *)) {
     SessionPropertyManager mgr(mSession);
     OSStatus rv = mgr.Set(kVTCompressionPropertyKey_ConstantBitRate,
                           AssertedCast<int32_t>(aBitsPerSec));
@@ -124,7 +130,7 @@ bool AppleVTEncoder::SetRealtime(bool aEnabled) {
     return false;
   }
 
-  if (__builtin_available(macos 11.0, *)) {
+  if (__builtin_available(macos 11.0, iOS 14.0, *)) {
     status = mgr.Set(
         kVTCompressionPropertyKey_PrioritizeEncodingSpeedOverQuality, aEnabled);
     LOGD("{} PrioritizeEncodingSpeedOverQuality, status: {}",
@@ -478,7 +484,7 @@ MediaResult AppleVTEncoder::InitSession() {
   }
 
   if (mConfig.mScalabilityMode != ScalabilityMode::None) {
-    if (__builtin_available(macos 11.3, *)) {
+    if (__builtin_available(macos 11.3, iOS 14.5, *)) {
       float baseLayerFPSRatio = 1.0f;
       switch (mConfig.mScalabilityMode) {
         case ScalabilityMode::L1T2:
@@ -543,11 +549,14 @@ MediaResult AppleVTEncoder::InitSession() {
     return colorSpaceResult;
   }
 
-  bool isUsingHW = false;
-  status =
-      mgr.Copy(kVTCompressionPropertyKey_UsingHardwareAcceleratedVideoEncoder,
-               isUsingHW);
-  mIsHardwareAccelerated = status == noErr && isUsingHW;
+  mIsHardwareAccelerated = false;
+  if (__builtin_available(macos 10.13, iOS 17.4, *)) {
+    bool isUsingHW = false;
+    status = mgr.Copy(
+        kVTCompressionPropertyKey_UsingHardwareAcceleratedVideoEncoder,
+        isUsingHW);
+    mIsHardwareAccelerated = status == noErr && isUsingHW;
+  }
   LOGD("Using hw acceleration: {}", mIsHardwareAccelerated ? "yes" : "no");
 
   errorExit.release();
diff --git a/dom/media/systemservices/objc_video_capture/device_info_objc.mm b/dom/media/systemservices/objc_video_capture/device_info_objc.mm
index 048565e0..f26a67da 100644
--- a/dom/media/systemservices/objc_video_capture/device_info_objc.mm
+++ b/dom/media/systemservices/objc_video_capture/device_info_objc.mm
@@ -16,6 +16,13 @@
 
 #import "device_info_objc.h"
 
+static BOOL CaptureDeviceIsSuspended(AVCaptureDevice* device) {
+  if (@available(iOS 14.0, macOS 10.15, *)) {
+    return [device isSuspended];
+  }
+  return NO;
+}
+
 @implementation DeviceInfoIosObjC
 
 - (id)init {
@@ -50,7 +57,7 @@
   @try {
     for (AVCaptureDevice* device in
          [AVCaptureDevice devicesWithMediaType:AVMediaTypeVideo]) {
-      if ([device isSuspended]) {
+      if (CaptureDeviceIsSuspended(device)) {
         continue;
       }
       cnt++;
@@ -66,7 +73,7 @@
   @try {
     for (AVCaptureDevice* device in
          [AVCaptureDevice devicesWithMediaType:AVMediaTypeVideo]) {
-      if ([device isSuspended]) {
+      if (CaptureDeviceIsSuspended(device)) {
         continue;
       }
       if (cnt == index) {
@@ -84,7 +91,7 @@
 + (AVCaptureDevice*)captureDeviceForUniqueId:(NSString*)uniqueId {
   for (AVCaptureDevice* device in
        [AVCaptureDevice devicesWithMediaType:AVMediaTypeVideo]) {
-    if ([device isSuspended]) {
+    if (CaptureDeviceIsSuspended(device)) {
       continue;
     }
     if ([uniqueId isEqual:device.uniqueID]) {
diff --git a/gfx/layers/NativeLayerCA.mm b/gfx/layers/NativeLayerCA.mm
index 230c7ac7..7689ed6c 100644
--- a/gfx/layers/NativeLayerCA.mm
+++ b/gfx/layers/NativeLayerCA.mm
@@ -542,8 +542,10 @@ void NativeLayerRootCA::SetWindowIsFullscreen(bool aFullscreen) {
 }
 
 /* static */ bool IsCGColorOpaqueBlack(CGColorRef aColor) {
-  if (CGColorEqualToColor(aColor, CGColorGetConstantColor(kCGColorBlack))) {
-    return true;
+  if (@available(macOS 10.13, iOS 14.0, *)) {
+    if (CGColorEqualToColor(aColor, CGColorGetConstantColor(kCGColorBlack))) {
+      return true;
+    }
   }
   size_t componentCount = CGColorGetNumberOfComponents(aColor);
   if (componentCount == 0) {
@@ -1583,8 +1585,7 @@ bool NativeLayerCARepresentation::EnqueueSurface(IOSurfaceRef aSurfaceRef) {
   AVSampleBufferDisplayLayer* videoLayer =
       (AVSampleBufferDisplayLayer*)mContentCALayer;
 
-  if ([videoLayer
-          respondsToSelector:@selector(requiresFlushToResumeDecoding)]) {
+  if (@available(macOS 11.0, iOS 14.0, *)) {
     if (videoLayer.requiresFlushToResumeDecoding) {
       [videoLayer flush];
     }
diff --git a/widget/uikit/ScreenHelperUIKit.mm b/widget/uikit/ScreenHelperUIKit.mm
index b773928a..4e865962 100644
--- a/widget/uikit/ScreenHelperUIKit.mm
+++ b/widget/uikit/ScreenHelperUIKit.mm
@@ -87,7 +87,7 @@ static already_AddRefed<Screen> MakeScreen(UIScreen* aScreen,
   // values and pretend the SDR content brightness is 100 which is mentioned in
   // Apple docs as typical.
   float displayHDRHeadroom = 1.0f;
-  if ([aScreen respondsToSelector:@selector(potentialEDRHeadroom)]) {
+  if (@available(iOS 16.0, *)) {
     displayHDRHeadroom = [aScreen potentialEDRHeadroom];
   }
   float sdrContentBrightness = 100.0f;
