diff --git a/gfx/layers/NativeLayerCA.mm b/gfx/layers/NativeLayerCA.mm
--- a/gfx/layers/NativeLayerCA.mm
+++ b/gfx/layers/NativeLayerCA.mm
@@ -542,8 +542,10 @@
 }
 
 /* static */ bool IsCGColorOpaqueBlack(CGColorRef aColor) {
-  if (CGColorEqualToColor(aColor, CGColorGetConstantColor(kCGColorBlack))) {
-    return true;
+  if (@available(macOS 10.13, iOS 14.0, *)) {
+    if (CGColorEqualToColor(aColor, CGColorGetConstantColor(kCGColorBlack))) {
+      return true;
+    }
   }
   size_t componentCount = CGColorGetNumberOfComponents(aColor);
   if (componentCount == 0) {
@@ -1309,7 +1311,7 @@
     NSLog(@"ColorSpace is %@.\n", colorSpace);
 #endif
 
-    if (@available(macOS 12.0, *)) {
+    if (@available(macOS 12.0, iOS 15.0, *)) {
       CFDictionaryRef bufferAttachments =
           CVBufferCopyAttachments(aBuffer, kCVAttachmentMode_ShouldPropagate);
       NSLog(@"Buffer attachments are %@.\n", bufferAttachments);
@@ -1596,8 +1598,7 @@
   AVSampleBufferDisplayLayer* videoLayer =
       (AVSampleBufferDisplayLayer*)mContentCALayer;
 
-  if ([videoLayer
-          respondsToSelector:@selector(requiresFlushToResumeDecoding)]) {
+  if (@available(macOS 11.0, iOS 14.0, *)) {
     if (videoLayer.requiresFlushToResumeDecoding) {
       [videoLayer flush];
     }
