diff --git a/js/src/wasm/WasmBuiltins.cpp b/js/src/wasm/WasmBuiltins.cpp
index c4387aaa2e70..480de62546ae 100644
--- a/js/src/wasm/WasmBuiltins.cpp
+++ b/js/src/wasm/WasmBuiltins.cpp
@@ -2351,8 +2351,10 @@ bool wasm::EnsureBuiltinThunksInitialized(
   }
 
   masm.executableCopy(thunks->codeBase);
-  memset(thunks->codeBase + masm.bytesNeeded(), 0,
-         allocSize - masm.bytesNeeded());
+    uint8_t* writableCodeBase = WritableJitAllocationFromExecutable(
+        thunks->codeBase, thunks->codeSize);
+    memset(writableCodeBase + masm.bytesNeeded(), 0,
+      allocSize - masm.bytesNeeded());
 
   masm.processCodeLabels(thunks->codeBase);
   PatchDebugSymbolicAccesses(thunks->codeBase, masm);
